{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "IfkNrcv6dlxI",
    "outputId": "fab579cc-68b6-4f59-a9d4-faa952a860da"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading...\n",
      "From (original): https://drive.google.com/uc?id=1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO\n",
      "From (redirected): https://drive.google.com/uc?id=1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO&confirm=t&uuid=b0313a00-d3ff-4621-ad1e-140e52ddff99\n",
      "To: /content/CRISPR.R1.fastq\n",
      "100% 2.67G/2.67G [00:27<00:00, 97.0MB/s]\n",
      "Downloading...\n",
      "From (original): https://drive.google.com/uc?id=1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS\n",
      "From (redirected): https://drive.google.com/uc?id=1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS&confirm=t&uuid=6b1381a6-9a2a-465f-9a94-d1325361b9e3\n",
      "To: /content/CRISPR.R2.fastq\n",
      "100% 2.67G/2.67G [00:26<00:00, 101MB/s]\n",
      "Downloading...\n",
      "From (original): https://drive.google.com/uc?id=1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar\n",
      "From (redirected): https://drive.google.com/uc?id=1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar&confirm=t&uuid=4eeb4c99-1321-4b98-bce6-61b107b14b45\n",
      "To: /content/MM10.tar.gz\n",
      "100% 3.91G/3.91G [00:39<00:00, 98.5MB/s]\n"
     ]
    }
   ],
   "source": [
    "###### CRISPR sample\n",
    "\n",
    "!gdown 1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO\n",
    "!gdown 1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS\n",
    "\n",
    "### To make thinngs faster for the student, here is the indexed mouse genome, mm10.\n",
    "### You have to donwload it and decompressed it, you will get both the fasta and the BWT'ed files. Note all have same prefix Mouse<.files>.\n",
    "### Next semester we will cover how specialized computers are often used to solve this aligmnet problems faster.\n",
    "\n",
    "!gdown 1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "EphxRjbJbVQi",
    "outputId": "01b555fd-421f-486a-9b5e-bdd7621c48ff"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM10/\n",
      "MM10/Mouse.ann\n",
      "MM10/Mouse.pac\n",
      "MM10/Mouse.fasta\n",
      "MM10/Mouse.bwt\n",
      "MM10/Mouse.sa\n",
      "MM10/Mouse.amb\n"
     ]
    }
   ],
   "source": [
    "#the tar command is used to sort numerous files in a single file\n",
    "!tar -xzvf MM10.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "ySCZLMFkQFQ6",
    "outputId": "09bf38a4-b2cb-4d36-85d4-82aada4845a0"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "Suggested packages:\n",
      "  samtools\n",
      "The following NEW packages will be installed:\n",
      "  bwa\n",
      "0 upgraded, 1 newly installed, 0 to remove and 49 not upgraded.\n",
      "Need to get 195 kB of archives.\n",
      "After this operation, 466 kB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bwa amd64 0.7.17-6 [195 kB]\n",
      "Fetched 195 kB in 1s (294 kB/s)\n",
      "Selecting previously unselected package bwa.\n",
      "(Reading database ... 123632 files and directories currently installed.)\n",
      "Preparing to unpack .../bwa_0.7.17-6_amd64.deb ...\n",
      "Unpacking bwa (0.7.17-6) ...\n",
      "Setting up bwa (0.7.17-6) ...\n",
      "Processing triggers for man-db (2.10.2-1) ...\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "The following additional packages will be installed:\n",
      "  libhts3 libhtscodecs2\n",
      "Suggested packages:\n",
      "  cwltool\n",
      "The following NEW packages will be installed:\n",
      "  libhts3 libhtscodecs2 samtools\n",
      "0 upgraded, 3 newly installed, 0 to remove and 49 not upgraded.\n",
      "Need to get 963 kB of archives.\n",
      "After this operation, 2,270 kB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
      "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
      "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
      "Fetched 963 kB in 1s (1,043 kB/s)\n",
      "Selecting previously unselected package libhtscodecs2:amd64.\n",
      "(Reading database ... 123649 files and directories currently installed.)\n",
      "Preparing to unpack .../libhtscodecs2_1.1.1-3_amd64.deb ...\n",
      "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
      "Selecting previously unselected package libhts3:amd64.\n",
      "Preparing to unpack .../libhts3_1.13+ds-2build1_amd64.deb ...\n",
      "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
      "Selecting previously unselected package samtools.\n",
      "Preparing to unpack .../samtools_1.13-4_amd64.deb ...\n",
      "Unpacking samtools (1.13-4) ...\n",
      "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
      "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
      "Setting up samtools (1.13-4) ...\n",
      "Processing triggers for man-db (2.10.2-1) ...\n",
      "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
      "\n",
      "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
      "\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "Suggested packages:\n",
      "  python3-numpy python3-matplotlib texlive-latex-recommended\n",
      "The following NEW packages will be installed:\n",
      "  bcftools\n",
      "0 upgraded, 1 newly installed, 0 to remove and 49 not upgraded.\n",
      "Need to get 697 kB of archives.\n",
      "After this operation, 2,360 kB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n",
      "Fetched 697 kB in 1s (907 kB/s)\n",
      "Selecting previously unselected package bcftools.\n",
      "(Reading database ... 123748 files and directories currently installed.)\n",
      "Preparing to unpack .../bcftools_1.13-1_amd64.deb ...\n",
      "Unpacking bcftools (1.13-1) ...\n",
      "Setting up bcftools (1.13-1) ...\n",
      "Processing triggers for man-db (2.10.2-1) ...\n"
     ]
    }
   ],
   "source": [
    "#this software is used for analysis and changing the file format\n",
    "!apt install bwa\n",
    "!apt install samtools\n",
    "!apt install bcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "id": "6M94QJp0luU6"
   },
   "outputs": [],
   "source": [
    "!samtools faidx MM10/Mouse.fasta chr2 > chr2.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "6NS4_q47Ma-b",
    "outputId": "ffac1d5b-6799-4ada-9a06-94292528188b"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 2.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=364226448, availableWord=37628116\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 62069712 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114669760 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161416816 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202961696 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239882848 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272694448 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301853424 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327765984 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350793104 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 205.80 seconds elapse.\n",
      "[bwa_index] Update BWT... 1.88 sec\n",
      "[bwa_index] Pack forward-only FASTA... 1.61 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 103.05 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index chr2.fasta\n",
      "[main] Real time: 332.632 sec; CPU: 314.968 sec\n"
     ]
    }
   ],
   "source": [
    "#This code creates index files for the reference genome allowing efficient alignment with BWA\n",
    "!bwa index chr2.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "ljhwCOhlMbZh",
    "outputId": "d9b35324-8fe9-4709-d0fe-497933a0fc62"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 184, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 234, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (238.14, 59.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.912 CPU sec, 39.591 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 67, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (212, 267, 364)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 668)\n",
      "[M::mem_pestat] mean and std.dev: (258.77, 97.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 820)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.999 CPU sec, 29.431 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 462)\n",
      "[M::mem_pestat] mean and std.dev: (248.92, 59.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 546)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.502 CPU sec, 30.428 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 162, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 228, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 425)\n",
      "[M::mem_pestat] mean and std.dev: (236.90, 57.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 500)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.106 CPU sec, 26.766 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (213, 285, 353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 633)\n",
      "[M::mem_pestat] mean and std.dev: (286.59, 102.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 773)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.642 CPU sec, 32.576 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 13, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (158, 245, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 593)\n",
      "[M::mem_pestat] mean and std.dev: (226.77, 79.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 738)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.239 CPU sec, 28.766 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 86, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 291)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 473)\n",
      "[M::mem_pestat] mean and std.dev: (236.91, 55.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.494 CPU sec, 36.175 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 26, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 246, 273)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (54, 419)\n",
      "[M::mem_pestat] mean and std.dev: (225.13, 55.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.666 CPU sec, 26.912 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 226, 271)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 421)\n",
      "[M::mem_pestat] mean and std.dev: (228.06, 53.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 496)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.543 CPU sec, 33.276 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 213, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 437)\n",
      "[M::mem_pestat] mean and std.dev: (234.71, 61.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.643 CPU sec, 28.223 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (187, 237, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 565)\n",
      "[M::mem_pestat] mean and std.dev: (244.09, 65.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.072 CPU sec, 33.128 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 578)\n",
      "[M::mem_pestat] mean and std.dev: (254.21, 63.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 701)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.349 CPU sec, 29.028 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 231, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 469)\n",
      "[M::mem_pestat] mean and std.dev: (237.83, 65.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 559)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.426 CPU sec, 31.427 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 242, 283)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (55, 435)\n",
      "[M::mem_pestat] mean and std.dev: (243.14, 53.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 511)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.426 CPU sec, 25.872 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 35, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 203, 256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 392)\n",
      "[M::mem_pestat] mean and std.dev: (215.48, 43.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 460)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.198 CPU sec, 31.353 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 20872, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 506)\n",
      "[M::mem_pestat] mean and std.dev: (263.84, 57.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 18.825 CPU sec, 19.134 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47226, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (13, 503)\n",
      "[M::mem_pestat] mean and std.dev: (263.62, 57.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.518 CPU sec, 14.849 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47220, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 507)\n",
      "[M::mem_pestat] mean and std.dev: (264.46, 57.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 13.962 CPU sec, 14.264 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47418, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 507)\n",
      "[M::mem_pestat] mean and std.dev: (264.51, 57.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 13.946 CPU sec, 14.274 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47964, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 506)\n",
      "[M::mem_pestat] mean and std.dev: (264.30, 57.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.326 CPU sec, 10.373 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 47962, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.66, 56.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 13.679 CPU sec, 13.730 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47721, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.49, 57.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.371 CPU sec, 14.982 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 45400, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 256, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 507)\n",
      "[M::mem_pestat] mean and std.dev: (264.20, 57.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.619 CPU sec, 15.091 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 46647, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 506)\n",
      "[M::mem_pestat] mean and std.dev: (264.10, 57.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.323 CPU sec, 12.682 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47269, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.14, 56.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.553 CPU sec, 12.623 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 47026, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.39, 57.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.307 CPU sec, 14.688 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47373, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (13, 503)\n",
      "[M::mem_pestat] mean and std.dev: (263.73, 57.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.310 CPU sec, 14.674 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47826, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.62, 57.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 17.241 CPU sec, 18.391 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44891, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (9, 509)\n",
      "[M::mem_pestat] mean and std.dev: (264.36, 57.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.878 CPU sec, 10.724 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46677, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (8, 508)\n",
      "[M::mem_pestat] mean and std.dev: (263.58, 57.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.189 CPU sec, 14.555 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41795, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (263.87, 57.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 18.677 CPU sec, 19.079 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 4123, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 259, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 510)\n",
      "[M::mem_pestat] mean and std.dev: (264.40, 57.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 610)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.116 CPU sec, 25.578 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 228, 262)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (79, 384)\n",
      "[M::mem_pestat] mean and std.dev: (237.65, 54.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (18, 456)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.447 CPU sec, 29.136 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 242, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 609)\n",
      "[M::mem_pestat] mean and std.dev: (233.15, 91.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 747)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.552 CPU sec, 29.153 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 33, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 587)\n",
      "[M::mem_pestat] mean and std.dev: (245.20, 64.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 716)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.710 CPU sec, 30.291 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 243, 326)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 576)\n",
      "[M::mem_pestat] mean and std.dev: (252.57, 64.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 701)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.891 CPU sec, 33.260 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (189, 228, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 447)\n",
      "[M::mem_pestat] mean and std.dev: (233.95, 58.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 533)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.774 CPU sec, 28.341 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 145, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 250, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (248.87, 69.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.247 CPU sec, 28.506 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 35, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 240, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 418)\n",
      "[M::mem_pestat] mean and std.dev: (228.82, 56.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.750 CPU sec, 30.744 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48, 0, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 248, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 540)\n",
      "[M::mem_pestat] mean and std.dev: (249.46, 51.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 651)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.501 CPU sec, 28.089 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 73, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 249, 338)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 612)\n",
      "[M::mem_pestat] mean and std.dev: (242.31, 65.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 749)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.804 CPU sec, 32.798 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 269)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (62, 407)\n",
      "[M::mem_pestat] mean and std.dev: (239.59, 47.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 476)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.526 CPU sec, 24.948 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 21, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 251, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 497)\n",
      "[M::mem_pestat] mean and std.dev: (253.90, 58.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.168 CPU sec, 31.332 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 236, 285)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 479)\n",
      "[M::mem_pestat] mean and std.dev: (238.64, 62.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.603 CPU sec, 26.146 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 136, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 249, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 519)\n",
      "[M::mem_pestat] mean and std.dev: (253.63, 63.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 625)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.279 CPU sec, 31.999 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 235, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 418)\n",
      "[M::mem_pestat] mean and std.dev: (224.86, 59.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 34.409 CPU sec, 35.787 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 222, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 491)\n",
      "[M::mem_pestat] mean and std.dev: (229.81, 50.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 588)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.144 CPU sec, 32.143 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 265, 1257)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3379)\n",
      "[M::mem_pestat] mean and std.dev: (695.85, 880.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4440)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 54.072 CPU sec, 55.577 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 79, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 233, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 538)\n",
      "[M::mem_pestat] mean and std.dev: (242.21, 63.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 649)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.851 CPU sec, 28.601 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 237, 1348)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3638)\n",
      "[M::mem_pestat] mean and std.dev: (625.19, 928.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4783)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 56.492 CPU sec, 57.831 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 65, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (220, 254, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 481)\n",
      "[M::mem_pestat] mean and std.dev: (249.14, 59.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.080 CPU sec, 25.786 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 33, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (224, 259, 748)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1796)\n",
      "[M::mem_pestat] mean and std.dev: (417.77, 368.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2320)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 40.456 CPU sec, 41.255 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 238, 284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 456)\n",
      "[M::mem_pestat] mean and std.dev: (234.60, 69.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 542)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.499 CPU sec, 27.923 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 581)\n",
      "[M::mem_pestat] mean and std.dev: (247.59, 71.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 705)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 34.014 CPU sec, 37.293 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 243, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 244, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 477)\n",
      "[M::mem_pestat] mean and std.dev: (245.08, 56.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.636 CPU sec, 29.058 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 107, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 250, 322)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 560)\n",
      "[M::mem_pestat] mean and std.dev: (251.70, 70.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 679)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.427 CPU sec, 25.929 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 40, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (206, 260, 328)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 572)\n",
      "[M::mem_pestat] mean and std.dev: (253.03, 59.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.043 CPU sec, 31.965 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 90, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (185, 218, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
      "[M::mem_pestat] mean and std.dev: (226.98, 68.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.121 CPU sec, 25.666 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 25, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 231, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 459)\n",
      "[M::mem_pestat] mean and std.dev: (243.68, 59.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.468 CPU sec, 29.430 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 256, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 548)\n",
      "[M::mem_pestat] mean and std.dev: (240.62, 53.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 664)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.958 CPU sec, 27.672 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 238, 738)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1810)\n",
      "[M::mem_pestat] mean and std.dev: (249.00, 112.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2346)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 44.412 CPU sec, 45.158 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 40, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 226, 274)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 422)\n",
      "[M::mem_pestat] mean and std.dev: (241.90, 59.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 496)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.233 CPU sec, 25.598 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 102, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 233, 346)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 638)\n",
      "[M::mem_pestat] mean and std.dev: (239.11, 69.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 784)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.708 CPU sec, 32.508 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 19, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (191, 235, 273)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 437)\n",
      "[M::mem_pestat] mean and std.dev: (227.00, 62.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 519)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.852 CPU sec, 27.308 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 416, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 235, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 485)\n",
      "[M::mem_pestat] mean and std.dev: (245.35, 65.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.659 CPU sec, 28.111 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 30, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 215, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 482)\n",
      "[M::mem_pestat] mean and std.dev: (236.00, 74.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.786 CPU sec, 33.766 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 254, 298)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 496)\n",
      "[M::mem_pestat] mean and std.dev: (241.85, 76.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.040 CPU sec, 29.271 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 57, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 362)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 686)\n",
      "[M::mem_pestat] mean and std.dev: (253.54, 70.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 848)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 38.638 CPU sec, 40.615 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 241, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
      "[M::mem_pestat] mean and std.dev: (258.70, 74.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 626)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.701 CPU sec, 27.784 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 277, 1367)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3691)\n",
      "[M::mem_pestat] mean and std.dev: (655.47, 674.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4853)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 55.658 CPU sec, 57.034 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 17, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (233, 286, 363)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 623)\n",
      "[M::mem_pestat] mean and std.dev: (255.29, 66.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 753)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.274 CPU sec, 32.117 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 26, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (213, 279, 1414)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3816)\n",
      "[M::mem_pestat] mean and std.dev: (702.12, 758.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5017)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 56.344 CPU sec, 60.854 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (183, 261, 1394)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3816)\n",
      "[M::mem_pestat] mean and std.dev: (688.23, 757.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5027)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 55.333 CPU sec, 56.231 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 68, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 237, 288)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 464)\n",
      "[M::mem_pestat] mean and std.dev: (239.67, 58.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 552)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.018 CPU sec, 27.573 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 225, 282)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 466)\n",
      "[M::mem_pestat] mean and std.dev: (234.64, 62.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 558)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.263 CPU sec, 32.096 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 282, 1785)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4941)\n",
      "[M::mem_pestat] mean and std.dev: (786.83, 887.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6519)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 62.225 CPU sec, 63.432 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 37, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 222, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 490)\n",
      "[M::mem_pestat] mean and std.dev: (237.70, 59.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 587)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.203 CPU sec, 30.021 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 83, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (230, 294, 376)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 668)\n",
      "[M::mem_pestat] mean and std.dev: (282.12, 98.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 814)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.199 CPU sec, 28.820 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 127, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 260, 338)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 600)\n",
      "[M::mem_pestat] mean and std.dev: (252.64, 71.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 731)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.642 CPU sec, 32.750 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 28, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (189, 210, 254)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (59, 384)\n",
      "[M::mem_pestat] mean and std.dev: (212.20, 48.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 449)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.877 CPU sec, 26.438 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 148, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 255, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 593)\n",
      "[M::mem_pestat] mean and std.dev: (249.18, 64.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 723)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 33.503 CPU sec, 34.605 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 350, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 234, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 458)\n",
      "[M::mem_pestat] mean and std.dev: (240.85, 58.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.464 CPU sec, 27.990 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 130, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 238, 334)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 616)\n",
      "[M::mem_pestat] mean and std.dev: (227.04, 77.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 757)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.216 CPU sec, 33.113 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 94, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 467)\n",
      "[M::mem_pestat] mean and std.dev: (249.47, 66.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.074 CPU sec, 26.695 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (186, 219, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
      "[M::mem_pestat] mean and std.dev: (221.72, 65.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 598)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.585 CPU sec, 29.073 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 219, 276)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 440)\n",
      "[M::mem_pestat] mean and std.dev: (218.87, 64.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 522)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.152 CPU sec, 30.932 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 271, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (264.93, 65.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 637)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.364 CPU sec, 26.852 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (159, 220, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 561)\n",
      "[M::mem_pestat] mean and std.dev: (209.25, 76.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 695)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.488 CPU sec, 32.484 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 63, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (212, 243, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 563)\n",
      "[M::mem_pestat] mean and std.dev: (243.98, 55.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 680)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.329 CPU sec, 26.866 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 67, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (215, 273, 408)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 794)\n",
      "[M::mem_pestat] mean and std.dev: (275.18, 114.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 987)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 35.553 CPU sec, 39.968 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 23, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 271, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 692)\n",
      "[M::mem_pestat] mean and std.dev: (263.45, 103.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 858)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.810 CPU sec, 26.040 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 77, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 292)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 476)\n",
      "[M::mem_pestat] mean and std.dev: (239.12, 61.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.079 CPU sec, 23.245 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 12, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 248, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 505)\n",
      "[M::mem_pestat] mean and std.dev: (242.33, 48.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.047 CPU sec, 23.362 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 386, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 250, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (3, 513)\n",
      "[M::mem_pestat] mean and std.dev: (257.69, 58.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 615)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.723 CPU sec, 24.969 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 64, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (197, 228, 300)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (236.80, 69.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.901 CPU sec, 25.099 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 40, 0, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 557)\n",
      "[M::mem_pestat] mean and std.dev: (225.45, 41.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 676)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.629 CPU sec, 25.951 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 240, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 425)\n",
      "[M::mem_pestat] mean and std.dev: (237.85, 53.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 500)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.354 CPU sec, 24.525 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 251, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 525)\n",
      "[M::mem_pestat] mean and std.dev: (251.62, 65.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 633)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.112 CPU sec, 25.354 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 38, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 339)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 617)\n",
      "[M::mem_pestat] mean and std.dev: (236.68, 75.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 756)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.369 CPU sec, 25.587 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 231, 284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (59, 434)\n",
      "[M::mem_pestat] mean and std.dev: (242.38, 56.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 509)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.653 CPU sec, 24.906 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 230, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (240.94, 64.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 672)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.009 CPU sec, 26.268 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (232, 268, 351)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 589)\n",
      "[M::mem_pestat] mean and std.dev: (263.09, 75.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 708)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.018 CPU sec, 25.238 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 78, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 233, 282)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 450)\n",
      "[M::mem_pestat] mean and std.dev: (233.42, 56.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.460 CPU sec, 25.795 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 77, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 324)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 572)\n",
      "[M::mem_pestat] mean and std.dev: (251.89, 75.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 696)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.481 CPU sec, 25.754 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 238, 285)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (33, 453)\n",
      "[M::mem_pestat] mean and std.dev: (235.13, 55.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 537)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.357 CPU sec, 23.616 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 239, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (32, 447)\n",
      "[M::mem_pestat] mean and std.dev: (236.52, 63.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 530)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.045 CPU sec, 24.243 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 224, 305)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 515)\n",
      "[M::mem_pestat] mean and std.dev: (234.22, 63.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.359 CPU sec, 24.609 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 268, 345)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 639)\n",
      "[M::mem_pestat] mean and std.dev: (261.27, 86.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 786)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.841 CPU sec, 27.116 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (220, 284, 1441)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3883)\n",
      "[M::mem_pestat] mean and std.dev: (713.05, 909.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5104)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 52.089 CPU sec, 52.763 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 38, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 239, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 499)\n",
      "[M::mem_pestat] mean and std.dev: (226.56, 70.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.470 CPU sec, 25.752 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 40, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 216, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 499)\n",
      "[M::mem_pestat] mean and std.dev: (234.69, 76.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.967 CPU sec, 25.189 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 77, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 230, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 549)\n",
      "[M::mem_pestat] mean and std.dev: (247.77, 80.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 665)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.543 CPU sec, 25.774 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 26, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (226, 254, 274)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (130, 370)\n",
      "[M::mem_pestat] mean and std.dev: (253.48, 37.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (82, 418)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.601 CPU sec, 32.522 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 137, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 240, 288)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 462)\n",
      "[M::mem_pestat] mean and std.dev: (243.04, 59.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.745 CPU sec, 24.920 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 233, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 517)\n",
      "[M::mem_pestat] mean and std.dev: (231.66, 65.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 625)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.806 CPU sec, 25.103 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 69, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 226, 290)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
      "[M::mem_pestat] mean and std.dev: (234.42, 74.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.989 CPU sec, 25.280 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 105, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 257, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (249.04, 64.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 649)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.975 CPU sec, 26.283 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (236, 309, 3383)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9677)\n",
      "[M::mem_pestat] mean and std.dev: (1517.14, 1751.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12824)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 96.392 CPU sec, 97.576 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 21, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (223, 241, 294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (81, 436)\n",
      "[M::mem_pestat] mean and std.dev: (250.10, 45.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (10, 507)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.323 CPU sec, 25.547 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 66, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 229, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 425)\n",
      "[M::mem_pestat] mean and std.dev: (233.60, 52.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 500)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.334 CPU sec, 22.476 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 5, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.308 CPU sec, 21.541 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (214, 325, 1627)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4453)\n",
      "[M::mem_pestat] mean and std.dev: (805.23, 931.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5866)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 56.193 CPU sec, 56.971 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 207, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
      "[M::mem_pestat] mean and std.dev: (218.67, 57.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.004 CPU sec, 26.201 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 178, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 542)\n",
      "[M::mem_pestat] mean and std.dev: (249.28, 71.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 656)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.611 CPU sec, 25.835 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 271, 323)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 553)\n",
      "[M::mem_pestat] mean and std.dev: (258.34, 83.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 668)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.828 CPU sec, 25.075 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 215, 268)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 424)\n",
      "[M::mem_pestat] mean and std.dev: (222.74, 54.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 502)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.441 CPU sec, 24.654 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 20, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 243, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 482)\n",
      "[M::mem_pestat] mean and std.dev: (245.00, 58.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 575)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.213 CPU sec, 24.376 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 294, 449)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 947)\n",
      "[M::mem_pestat] mean and std.dev: (254.27, 83.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1196)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.413 CPU sec, 28.700 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 104, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 239, 299)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 507)\n",
      "[M::mem_pestat] mean and std.dev: (242.38, 72.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.811 CPU sec, 25.089 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 20, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 234, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 559)\n",
      "[M::mem_pestat] mean and std.dev: (242.15, 72.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 680)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.113 CPU sec, 25.306 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 68, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (185, 220, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
      "[M::mem_pestat] mean and std.dev: (213.68, 66.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.792 CPU sec, 26.044 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 21, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 210, 263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (74, 389)\n",
      "[M::mem_pestat] mean and std.dev: (222.60, 52.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (11, 452)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.481 CPU sec, 25.806 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 58, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 230, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 594)\n",
      "[M::mem_pestat] mean and std.dev: (224.96, 62.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 726)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 33.545 CPU sec, 36.044 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 27, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (216, 251, 272)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (104, 384)\n",
      "[M::mem_pestat] mean and std.dev: (239.23, 49.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (40, 440)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.862 CPU sec, 25.119 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 171, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 251, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 515)\n",
      "[M::mem_pestat] mean and std.dev: (246.57, 58.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 619)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.316 CPU sec, 25.582 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 51, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 233, 280)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (40, 440)\n",
      "[M::mem_pestat] mean and std.dev: (243.45, 68.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 520)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.360 CPU sec, 24.579 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (218, 265, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 467)\n",
      "[M::mem_pestat] mean and std.dev: (269.11, 68.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 550)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.407 CPU sec, 23.535 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 248, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 549)\n",
      "[M::mem_pestat] mean and std.dev: (249.86, 58.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 664)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.562 CPU sec, 24.819 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 38, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (191, 206, 273)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 437)\n",
      "[M::mem_pestat] mean and std.dev: (221.88, 53.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 519)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.399 CPU sec, 24.541 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 65, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 247, 300)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 500)\n",
      "[M::mem_pestat] mean and std.dev: (235.04, 56.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.038 CPU sec, 25.205 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 230, 268)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (64, 404)\n",
      "[M::mem_pestat] mean and std.dev: (236.94, 55.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 472)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.881 CPU sec, 24.026 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 66, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 254, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1410)\n",
      "[M::mem_pestat] mean and std.dev: (305.57, 206.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1813)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.644 CPU sec, 32.902 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 488)\n",
      "[M::mem_pestat] mean and std.dev: (245.24, 63.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.579 CPU sec, 24.797 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 227, 284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 462)\n",
      "[M::mem_pestat] mean and std.dev: (225.48, 60.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.252 CPU sec, 24.496 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 69, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 246, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 548)\n",
      "[M::mem_pestat] mean and std.dev: (241.30, 58.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 664)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.730 CPU sec, 25.936 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 112, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 244, 291)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 473)\n",
      "[M::mem_pestat] mean and std.dev: (239.08, 62.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.093 CPU sec, 24.270 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 226, 277)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (70, 415)\n",
      "[M::mem_pestat] mean and std.dev: (236.00, 56.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 484)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.797 CPU sec, 25.030 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 17, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (211, 284, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 529)\n",
      "[M::mem_pestat] mean and std.dev: (256.40, 56.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 635)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.689 CPU sec, 27.227 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 10, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (182, 221, 243)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (60, 365)\n",
      "[M::mem_pestat] mean and std.dev: (205.78, 35.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 426)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.976 CPU sec, 24.185 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 358, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 247, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 485)\n",
      "[M::mem_pestat] mean and std.dev: (248.39, 64.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.742 CPU sec, 25.030 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 78, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 229, 274)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 426)\n",
      "[M::mem_pestat] mean and std.dev: (224.29, 57.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 502)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.932 CPU sec, 23.116 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 201, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 433)\n",
      "[M::mem_pestat] mean and std.dev: (214.17, 60.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.064 CPU sec, 24.245 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (183, 214, 241)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (67, 357)\n",
      "[M::mem_pestat] mean and std.dev: (216.50, 55.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 439)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.047 CPU sec, 24.172 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (256.74, 59.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 652)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.467 CPU sec, 25.697 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 760, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 250, 304)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 502)\n",
      "[M::mem_pestat] mean and std.dev: (259.69, 61.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.556 CPU sec, 24.784 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 125, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 268, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 597)\n",
      "[M::mem_pestat] mean and std.dev: (253.15, 74.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 729)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.018 CPU sec, 26.771 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 227, 276)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (75, 410)\n",
      "[M::mem_pestat] mean and std.dev: (235.16, 55.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (8, 477)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 36.320 CPU sec, 40.274 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 158, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 254, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (242.08, 66.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 637)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.442 CPU sec, 24.591 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 174, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 241, 306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 514)\n",
      "[M::mem_pestat] mean and std.dev: (255.11, 71.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 618)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.346 CPU sec, 25.586 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 250, 322)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 564)\n",
      "[M::mem_pestat] mean and std.dev: (251.40, 62.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.058 CPU sec, 25.374 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 530)\n",
      "[M::mem_pestat] mean and std.dev: (250.23, 62.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 640)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.683 CPU sec, 24.886 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 64, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 292)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 476)\n",
      "[M::mem_pestat] mean and std.dev: (247.98, 58.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.239 CPU sec, 25.351 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 244, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (241.42, 56.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.986 CPU sec, 25.215 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (214, 247, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 514)\n",
      "[M::mem_pestat] mean and std.dev: (246.97, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 614)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.023 CPU sec, 24.135 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 437)\n",
      "[M::mem_pestat] mean and std.dev: (241.49, 63.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.663 CPU sec, 24.800 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 236, 311)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (250.02, 70.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 635)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.347 CPU sec, 26.653 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 232, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (66, 421)\n",
      "[M::mem_pestat] mean and std.dev: (243.35, 61.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.348 CPU sec, 24.564 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 222, 282)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 450)\n",
      "[M::mem_pestat] mean and std.dev: (229.90, 65.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.679 CPU sec, 24.963 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 278)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (44, 434)\n",
      "[M::mem_pestat] mean and std.dev: (241.82, 53.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 512)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.066 CPU sec, 24.274 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 212, 252)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (96, 356)\n",
      "[M::mem_pestat] mean and std.dev: (217.39, 54.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 435)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.081 CPU sec, 23.350 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 22, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (141, 200, 239)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 435)\n",
      "[M::mem_pestat] mean and std.dev: (199.00, 57.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 533)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.915 CPU sec, 24.103 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 33, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (225, 266, 366)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 648)\n",
      "[M::mem_pestat] mean and std.dev: (274.29, 94.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 789)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.304 CPU sec, 25.527 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 89, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 458)\n",
      "[M::mem_pestat] mean and std.dev: (240.88, 61.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.419 CPU sec, 25.633 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 271, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 670)\n",
      "[M::mem_pestat] mean and std.dev: (260.72, 87.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 825)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.869 CPU sec, 26.147 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 103, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 237, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 491)\n",
      "[M::mem_pestat] mean and std.dev: (238.83, 62.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 588)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.110 CPU sec, 24.302 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 59, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 235, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (233.81, 70.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.031 CPU sec, 25.179 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (179, 235, 386)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 800)\n",
      "[M::mem_pestat] mean and std.dev: (215.79, 108.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1007)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.693 CPU sec, 26.926 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 275, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 584)\n",
      "[M::mem_pestat] mean and std.dev: (253.53, 64.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 711)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.780 CPU sec, 25.902 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 226, 256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (94, 364)\n",
      "[M::mem_pestat] mean and std.dev: (233.80, 54.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (15, 452)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.130 CPU sec, 24.338 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (223, 264, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 493)\n",
      "[M::mem_pestat] mean and std.dev: (249.05, 46.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.631 CPU sec, 34.984 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 100, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 256, 304)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (4, 504)\n",
      "[M::mem_pestat] mean and std.dev: (249.59, 63.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.943 CPU sec, 25.125 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 237, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 475)\n",
      "[M::mem_pestat] mean and std.dev: (233.37, 56.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.855 CPU sec, 24.118 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 250, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 255, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (253.96, 65.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.103 CPU sec, 24.281 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 62, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (8, 488)\n",
      "[M::mem_pestat] mean and std.dev: (240.73, 70.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 584)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.788 CPU sec, 25.941 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 242, 306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 512)\n",
      "[M::mem_pestat] mean and std.dev: (244.71, 67.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 615)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.699 CPU sec, 25.903 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 232, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 486)\n",
      "[M::mem_pestat] mean and std.dev: (238.19, 57.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.007 CPU sec, 25.163 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 26, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 249, 292)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 476)\n",
      "[M::mem_pestat] mean and std.dev: (233.00, 71.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.661 CPU sec, 24.878 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 228, 290)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 472)\n",
      "[M::mem_pestat] mean and std.dev: (239.80, 63.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 563)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.698 CPU sec, 23.917 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 132, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 249, 312)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (250.65, 64.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.206 CPU sec, 25.375 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 140, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 238, 325)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 575)\n",
      "[M::mem_pestat] mean and std.dev: (230.57, 57.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 700)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.908 CPU sec, 26.171 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44, 0, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 226, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 437)\n",
      "[M::mem_pestat] mean and std.dev: (239.45, 61.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.580 CPU sec, 24.758 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 70, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 524)\n",
      "[M::mem_pestat] mean and std.dev: (242.09, 61.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 632)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.446 CPU sec, 26.733 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 227, 276)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 428)\n",
      "[M::mem_pestat] mean and std.dev: (238.18, 50.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 504)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.259 CPU sec, 24.382 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 239, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (77, 392)\n",
      "[M::mem_pestat] mean and std.dev: (230.71, 47.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (14, 455)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.011 CPU sec, 24.176 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (230, 257, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 491)\n",
      "[M::mem_pestat] mean and std.dev: (256.43, 57.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.829 CPU sec, 27.386 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 252, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (3, 493)\n",
      "[M::mem_pestat] mean and std.dev: (237.72, 60.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 591)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.626 CPU sec, 24.816 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 114, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 248, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 525)\n",
      "[M::mem_pestat] mean and std.dev: (239.28, 69.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 633)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.108 CPU sec, 25.334 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 252, 294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (15, 480)\n",
      "[M::mem_pestat] mean and std.dev: (249.54, 51.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.757 CPU sec, 24.967 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 246, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 490)\n",
      "[M::mem_pestat] mean and std.dev: (245.64, 62.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 587)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.732 CPU sec, 24.917 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 106, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 479)\n",
      "[M::mem_pestat] mean and std.dev: (236.56, 58.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.447 CPU sec, 25.595 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 19, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 258, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (246.58, 77.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.183 CPU sec, 24.342 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 303, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 241, 298)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 502)\n",
      "[M::mem_pestat] mean and std.dev: (244.19, 68.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.702 CPU sec, 24.840 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 112, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 254, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 584)\n",
      "[M::mem_pestat] mean and std.dev: (254.82, 69.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 711)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.928 CPU sec, 26.134 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 253, 353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 641)\n",
      "[M::mem_pestat] mean and std.dev: (263.56, 94.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 785)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.436 CPU sec, 34.877 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 92, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 223, 278)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (38, 438)\n",
      "[M::mem_pestat] mean and std.dev: (229.08, 54.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 518)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.845 CPU sec, 25.009 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 164, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 255, 342)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 626)\n",
      "[M::mem_pestat] mean and std.dev: (251.41, 82.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 768)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.329 CPU sec, 27.563 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 115, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 236, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 509)\n",
      "[M::mem_pestat] mean and std.dev: (236.29, 59.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.398 CPU sec, 26.655 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 173, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 231, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 467)\n",
      "[M::mem_pestat] mean and std.dev: (240.95, 59.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.701 CPU sec, 27.876 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 100, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 485)\n",
      "[M::mem_pestat] mean and std.dev: (240.90, 61.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.434 CPU sec, 25.659 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 132, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 238, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 508)\n",
      "[M::mem_pestat] mean and std.dev: (244.71, 67.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.922 CPU sec, 26.084 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 160, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 247, 299)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 495)\n",
      "[M::mem_pestat] mean and std.dev: (249.74, 62.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.965 CPU sec, 26.214 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 85, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 224, 264)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (69, 394)\n",
      "[M::mem_pestat] mean and std.dev: (236.35, 56.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (4, 461)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.303 CPU sec, 24.459 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 93, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 203, 254)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 386)\n",
      "[M::mem_pestat] mean and std.dev: (220.40, 49.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 452)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.171 CPU sec, 24.249 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 72, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 230, 280)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 438)\n",
      "[M::mem_pestat] mean and std.dev: (241.07, 56.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 517)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.839 CPU sec, 24.923 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 240, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (116, 381)\n",
      "[M::mem_pestat] mean and std.dev: (250.47, 46.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (62, 438)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.150 CPU sec, 24.285 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1701, 10, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 243, 285)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (36, 451)\n",
      "[M::mem_pestat] mean and std.dev: (249.99, 49.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (610, 700, 717)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (396, 931)\n",
      "[M::mem_pestat] mean and std.dev: (667.60, 59.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (289, 1038)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.388 CPU sec, 22.501 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (255, 26456, 728, 134)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5935, 7042, 9226)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15808)\n",
      "[M::mem_pestat] mean and std.dev: (7037.24, 1985.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19099)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 244, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 454)\n",
      "[M::mem_pestat] mean and std.dev: (248.86, 47.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 538)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (853, 4550, 8569)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 24001)\n",
      "[M::mem_pestat] mean and std.dev: (4486.34, 3465.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 31717)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5829, 8417, 8805)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14757)\n",
      "[M::mem_pestat] mean and std.dev: (6357.01, 3083.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18693)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 7.309 CPU sec, 7.196 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 8324, 559, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5685, 5708, 5740)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5575, 5850)\n",
      "[M::mem_pestat] mean and std.dev: (5694.64, 45.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5513, 5905)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 244, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (31, 461)\n",
      "[M::mem_pestat] mean and std.dev: (246.99, 47.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 547)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (689, 767, 3145)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8057)\n",
      "[M::mem_pestat] mean and std.dev: (1593.57, 1394.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10513)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (9310, 9325, 9378)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (9174, 9514)\n",
      "[M::mem_pestat] mean and std.dev: (9336.54, 39.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (9106, 9582)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 52.532 CPU sec, 52.817 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.626 CPU sec, 11.586 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.892 CPU sec, 12.939 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 5, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 13.291 CPU sec, 13.316 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 5, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.797 CPU sec, 11.790 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 298, 10, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 255, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 583)\n",
      "[M::mem_pestat] mean and std.dev: (261.42, 72.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 710)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (276, 717, 1063)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2637)\n",
      "[M::mem_pestat] mean and std.dev: (669.67, 625.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3424)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.433 CPU sec, 21.552 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 6295, 203, 22)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (4679, 5916, 8665)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16637)\n",
      "[M::mem_pestat] mean and std.dev: (5639.33, 2488.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 20623)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 300, 871)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2169)\n",
      "[M::mem_pestat] mean and std.dev: (398.91, 382.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2818)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1592, 4042, 6612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16652)\n",
      "[M::mem_pestat] mean and std.dev: (4318.67, 3002.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21672)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1171, 2855, 7353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 19717)\n",
      "[M::mem_pestat] mean and std.dev: (4050.59, 3183.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 25899)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 33.433 CPU sec, 33.651 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (77, 41597, 1215, 96)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (2483, 5461, 7319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16991)\n",
      "[M::mem_pestat] mean and std.dev: (5062.60, 2795.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21827)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (227, 311, 1017)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2597)\n",
      "[M::mem_pestat] mean and std.dev: (499.95, 528.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3387)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1560, 3597, 6164)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15372)\n",
      "[M::mem_pestat] mean and std.dev: (3993.12, 2800.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19976)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1878, 4388, 7563)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 18933)\n",
      "[M::mem_pestat] mean and std.dev: (4636.20, 3018.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 24618)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 14.856 CPU sec, 14.920 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (26, 12969, 449, 27)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1055, 3323, 7046)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 19028)\n",
      "[M::mem_pestat] mean and std.dev: (4029.31, 3163.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 25019)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (227, 309, 968)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2450)\n",
      "[M::mem_pestat] mean and std.dev: (476.39, 487.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3191)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1429, 3340, 5771)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14455)\n",
      "[M::mem_pestat] mean and std.dev: (3873.27, 2791.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18797)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1677, 4695, 7916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 20394)\n",
      "[M::mem_pestat] mean and std.dev: (4646.81, 3049.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 26633)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.738 CPU sec, 32.933 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 350, 15, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 249, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (250.10, 68.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 672)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (164, 340, 1511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4205)\n",
      "[M::mem_pestat] mean and std.dev: (767.80, 869.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5552)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.688 CPU sec, 24.784 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 429, 29, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 251, 326)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 578)\n",
      "[M::mem_pestat] mean and std.dev: (242.62, 77.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 704)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (401, 1006, 1384)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3350)\n",
      "[M::mem_pestat] mean and std.dev: (912.25, 589.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4333)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 44.877 CPU sec, 45.052 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 309, 11, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 356)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 668)\n",
      "[M::mem_pestat] mean and std.dev: (253.90, 90.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 824)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (278, 1573, 6371)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 18557)\n",
      "[M::mem_pestat] mean and std.dev: (2603.55, 2806.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 24650)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 33.235 CPU sec, 36.757 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 393, 11, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 251, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 530)\n",
      "[M::mem_pestat] mean and std.dev: (245.13, 61.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 640)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (187, 337, 505)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1141)\n",
      "[M::mem_pestat] mean and std.dev: (285.22, 126.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1459)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.754 CPU sec, 24.901 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 352, 19, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 337)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 611)\n",
      "[M::mem_pestat] mean and std.dev: (248.12, 74.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 748)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (279, 806, 1684)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4494)\n",
      "[M::mem_pestat] mean and std.dev: (822.61, 613.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5899)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 54.904 CPU sec, 55.574 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 215, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 254, 336)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 608)\n",
      "[M::mem_pestat] mean and std.dev: (253.20, 73.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 744)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.616 CPU sec, 28.376 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 289, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 248, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 534)\n",
      "[M::mem_pestat] mean and std.dev: (253.85, 69.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 644)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.918 CPU sec, 25.255 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 381, 27, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 245, 305)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 515)\n",
      "[M::mem_pestat] mean and std.dev: (243.43, 65.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (393, 846, 1527)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3795)\n",
      "[M::mem_pestat] mean and std.dev: (968.20, 843.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4929)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 45.215 CPU sec, 45.771 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 428, 26, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 248, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 542)\n",
      "[M::mem_pestat] mean and std.dev: (255.63, 70.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 656)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 373, 905)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2323)\n",
      "[M::mem_pestat] mean and std.dev: (548.32, 509.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3032)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 36.940 CPU sec, 37.470 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 257, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 231, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 479)\n",
      "[M::mem_pestat] mean and std.dev: (243.10, 65.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.360 CPU sec, 23.514 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 398, 22, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 524)\n",
      "[M::mem_pestat] mean and std.dev: (248.99, 67.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 632)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 645, 1356)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3672)\n",
      "[M::mem_pestat] mean and std.dev: (559.95, 479.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4830)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 42.016 CPU sec, 42.362 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 251, 9, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (253.26, 67.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 652)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.185 CPU sec, 23.423 real sec\n",
      "[M::process] read 12220 sequences (1222000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 597, 35, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 241, 306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
      "[M::mem_pestat] mean and std.dev: (242.73, 66.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 627)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (239, 311, 882)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2168)\n",
      "[M::mem_pestat] mean and std.dev: (542.88, 453.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2811)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 35.884 CPU sec, 36.383 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 12220 reads in 1.922 CPU sec, 1.874 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem chr2.fasta CRISPR.R1.fastq CRISPR.R2.fastq\n",
      "[main] Real time: 6692.945 sec; CPU: 6550.567 sec\n"
     ]
    }
   ],
   "source": [
    "#This code aligns CRISPR sequencing reads to the reference genome using BWA's mem algorithm and saves the alignments in the SAM file\n",
    "!bwa mem chr2.fasta CRISPR.R1.fastq CRISPR.R2.fastq > aligned_reads_chr2.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "WsbqxV2Ll8-Z",
    "outputId": "ccc721fe-9dae-4c30-bce4-3b1f390ba2e3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 7 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "#Converting to a BAM file and then sorting reads\n",
    "!samtools view -Sb aligned_reads_chr2.sam > aligned_reads_chr2.bam\n",
    "!samtools sort aligned_reads_chr2.bam -o sorted_reads_chr2.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "id": "UJYaUY6HlKer"
   },
   "outputs": [],
   "source": [
    "#Indexing the file\n",
    "!samtools index sorted_reads_chr2.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "8zSpv1f0fdO8",
    "outputId": "1794dbf1-45c3-41dd-9525-2faeb3904fab"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n"
     ]
    }
   ],
   "source": [
    "#Identified genetic variations compared to the reference genome\n",
    "\n",
    "!bcftools mpileup -Ou -f chr2.fasta sorted_reads_chr2.bam | bcftools call -mv -Oz -o variants_chr2.vcf.gz #identify potential variants (SNPs, indels) in your aligned reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "id": "GuXz0I7-geum"
   },
   "outputs": [],
   "source": [
    "#Removing low-quality reads to only include high quality ones\n",
    "!bcftools filter -i 'QUAL>20 && DP>10' variants_chr2.vcf.gz -Oz -o filtered_variants_chr2.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "5BkKRdwlgiW5",
    "outputId": "c07475da-f4d3-4fd3-ac72-911c39718433"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting cyvcf2\n",
      "  Downloading cyvcf2-0.31.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.5 kB)\n",
      "Requirement already satisfied: numpy>=1.16.0 in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (1.26.4)\n",
      "Collecting coloredlogs (from cyvcf2)\n",
      "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n",
      "Requirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (8.1.7)\n",
      "Collecting humanfriendly>=9.1 (from coloredlogs->cyvcf2)\n",
      "  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n",
      "Downloading cyvcf2-0.31.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.8 MB)\n",
      "\u001b[2K   \u001b[90m\u001b[0m \u001b[32m6.8/6.8 MB\u001b[0m \u001b[31m41.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
      "\u001b[2K   \u001b[90m\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
      "\u001b[2K   \u001b[90m\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m5.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: humanfriendly, coloredlogs, cyvcf2\n",
      "Successfully installed coloredlogs-15.0.1 cyvcf2-0.31.1 humanfriendly-10.0\n"
     ]
    }
   ],
   "source": [
    "#installs the  library cyvcf2 which is used for parsing and working with VCF\n",
    "!pip install cyvcf2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "id": "9GeXpW7egqP2"
   },
   "outputs": [],
   "source": [
    "!bcftools view variants_chr2.vcf.gz -Ov > filtered_variants_chr2.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 411
    },
    "id": "YNkKlilxgunC",
    "outputId": "d038f698-71ba-4ebb-9391-7f710f4ead2e"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"df\",\n  \"rows\": 70683,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54986788,\n        \"min\": 3051590,\n        \"max\": 182002086,\n        \"num_unique_values\": 70674,\n        \"samples\": [\n          84317243\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 482,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 545,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 55.72852422128099,\n        \"min\": 3.0108699798583984,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 29713,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-9f8105ca-7a5a-45d6-a135-a6a771f6922f\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>52.203800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>45.253502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.249500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>55.332600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>59.256699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70678</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70679</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70680</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70681</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70682</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>5.826790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70683 rows  8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-9f8105ca-7a5a-45d6-a135-a6a771f6922f')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-9f8105ca-7a5a-45d6-a135-a6a771f6922f button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-9f8105ca-7a5a-45d6-a135-a6a771f6922f');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-d047964b-8925-4548-9fc4-a48fb80dbdac\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-d047964b-8925-4548-9fc4-a48fb80dbdac')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-d047964b-8925-4548-9fc4-a48fb80dbdac button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_5c59437d-19e2-42ea-90bd-9540bd979155\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_5c59437d-19e2-42ea-90bd-9540bd979155 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF ALT       QUAL  FILTER  GENOTYPE\n",
       "0      chr2    3051590 NaN   T   C  52.203800     NaN         1\n",
       "1      chr2    3051604 NaN   A   C  45.253502     NaN         1\n",
       "2      chr2    3051621 NaN   T   C  47.249500     NaN         1\n",
       "3      chr2    3051632 NaN   A   G  55.332600     NaN         1\n",
       "4      chr2    3051635 NaN   A   G  59.256699     NaN         1\n",
       "...     ...        ...  ..  ..  ..        ...     ...       ...\n",
       "70678  chr2  182002062 NaN   A   T  30.957300     NaN         1\n",
       "70679  chr2  182002065 NaN   G   A  40.932098     NaN         3\n",
       "70680  chr2  182002067 NaN   G   A  40.932098     NaN         3\n",
       "70681  chr2  182002068 NaN   A   T  30.957300     NaN         1\n",
       "70682  chr2  182002086 NaN   T   A   5.826790     NaN         1\n",
       "\n",
       "[70683 rows x 8 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this data shows the genotypes that are present as well as the comparison between the alternate mouse genome and the reference genome\n",
    "import cyvcf2\n",
    "import pandas as pd\n",
    "\n",
    "vcf_path = 'filtered_variants_chr2.vcf'\n",
    "\n",
    "# Converting VCF to TSV\n",
    "with open('output.tsv', 'w') as file:\n",
    "    for variant in cyvcf2.VCF(vcf_path):\n",
    "        genotype = variant.gt_types[0]\n",
    "        data_string = f\"{variant.CHROM}\\t{variant.POS}\\t{variant.ID}\\t{variant.REF}\\t{','.join(variant.ALT)}\\t{variant.QUAL}\\t{variant.FILTER}\\t{genotype}\\n\"\n",
    "        file.write(data_string)\n",
    "\n",
    "# Load TSV file into New DataFrame\n",
    "columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'GENOTYPE']\n",
    "df = pd.read_csv(\"output.tsv\", delimiter=\"\\t\", header=None, names=columns)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 411
    },
    "id": "X64wJMk0gvbF",
    "outputId": "9c42c981-2321-41aa-ba1b-67e73ca7cc60"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"filtered_df\",\n  \"rows\": 17761,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57461582,\n        \"min\": 3051696,\n        \"max\": 182002048,\n        \"num_unique_values\": 17759,\n        \"samples\": [\n          55775774\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 225,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 265,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54.77611585545069,\n        \"min\": 60.00130081176758,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 13556,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "filtered_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-a15c7f53-746b-4971-8502-cb6c0bff549d\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051696</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TGTCC</td>\n",
       "      <td>96.563599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>111.313004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052386</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>129.570007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052401</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>131.856003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>84.802002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70671</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>95.972900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70672</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>71.400002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70673</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>65.766403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70675</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>77.494499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70677</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>91.032799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17761 rows  8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a15c7f53-746b-4971-8502-cb6c0bff549d')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-a15c7f53-746b-4971-8502-cb6c0bff549d button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-a15c7f53-746b-4971-8502-cb6c0bff549d');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-72255cd1-ed10-4f36-aedb-42798ea3d365\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-72255cd1-ed10-4f36-aedb-42798ea3d365')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-72255cd1-ed10-4f36-aedb-42798ea3d365 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_98cbc0f9-b216-4f3d-bd1f-43d290aa5a83\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('filtered_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_98cbc0f9-b216-4f3d-bd1f-43d290aa5a83 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('filtered_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF    ALT        QUAL  FILTER  GENOTYPE\n",
       "6      chr2    3051696 NaN   T  TGTCC   96.563599     NaN         1\n",
       "29     chr2    3052361 NaN   G      T  111.313004     NaN         1\n",
       "31     chr2    3052386 NaN   T      G  129.570007     NaN         1\n",
       "32     chr2    3052401 NaN   C      G  131.856003     NaN         1\n",
       "33     chr2    3052429 NaN   A      G   84.802002     NaN         1\n",
       "...     ...        ...  ..  ..    ...         ...     ...       ...\n",
       "70671  chr2  182002007 NaN   A      G   95.972900     NaN         1\n",
       "70672  chr2  182002013 NaN   T      C   71.400002     NaN         1\n",
       "70673  chr2  182002021 NaN   G      A   65.766403     NaN         1\n",
       "70675  chr2  182002039 NaN   A      G   77.494499     NaN         1\n",
       "70677  chr2  182002048 NaN   T      C   91.032799     NaN         1\n",
       "\n",
       "[17761 rows x 8 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#converts the 'QUAL' column in a pandas DataFrame df to numeric values, filters the DataFrame to include only rows where the 'QUAL' value is greater than 60, and displays the filtered DataFrame\n",
    "#table shows the filtered dataframe from the previous table\n",
    "\n",
    "df['QUAL'] = pd.to_numeric(df['QUAL'], errors='coerce')\n",
    "\n",
    "filtered_df = df[df['QUAL'] > 60]\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 355
    },
    "id": "RVSuRwlwh-8I",
    "outputId": "ab78eceb-c45c-41db-d378-8b5eb3ca183a"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-16-47790675b30c>:6: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  intended_mutations_df = filtered_df[df['POS'].isin(intended_sites)]\n"
     ]
    },
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"intended_mutations_df\",\n  \"rows\": 9,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 21704765,\n        \"min\": 36937210,\n        \"max\": 87049235,\n        \"num_unique_values\": 9,\n        \"samples\": [\n          86658391\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 6,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 51.23562662600714,\n        \"min\": 102.33000183105467,\n        \"max\": 228.25,\n        \"num_unique_values\": 9,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "intended_mutations_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-a3fcaebe-1d45-48f3-b3d6-c55424d474eb\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16660</th>\n",
       "      <td>chr2</td>\n",
       "      <td>36937210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>194.453003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16667</th>\n",
       "      <td>chr2</td>\n",
       "      <td>36996899</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTTTTT</td>\n",
       "      <td>CTTTTTT</td>\n",
       "      <td>185.158005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33179</th>\n",
       "      <td>chr2</td>\n",
       "      <td>85400441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>228.171005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33346</th>\n",
       "      <td>chr2</td>\n",
       "      <td>85776687</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>115.749001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33665</th>\n",
       "      <td>chr2</td>\n",
       "      <td>85918029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.149002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34023</th>\n",
       "      <td>chr2</td>\n",
       "      <td>86198668</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>228.250000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34039</th>\n",
       "      <td>chr2</td>\n",
       "      <td>86236802</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CT</td>\n",
       "      <td>CTT</td>\n",
       "      <td>228.218994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34510</th>\n",
       "      <td>chr2</td>\n",
       "      <td>86658391</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>174.298996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34657</th>\n",
       "      <td>chr2</td>\n",
       "      <td>87049235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>102.330002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a3fcaebe-1d45-48f3-b3d6-c55424d474eb')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-a3fcaebe-1d45-48f3-b3d6-c55424d474eb button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-a3fcaebe-1d45-48f3-b3d6-c55424d474eb');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-68f38c2d-cf56-41b4-8c50-80f4e5d34b22\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-68f38c2d-cf56-41b4-8c50-80f4e5d34b22')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-68f38c2d-cf56-41b4-8c50-80f4e5d34b22 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_f47ee019-93af-42ff-8bcf-2a2fb713d229\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('intended_mutations_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_f47ee019-93af-42ff-8bcf-2a2fb713d229 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('intended_mutations_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM       POS  ID     REF      ALT        QUAL  FILTER  GENOTYPE\n",
       "16660  chr2  36937210 NaN       C        T  194.453003     NaN         1\n",
       "16667  chr2  36996899 NaN  CTTTTT  CTTTTTT  185.158005     NaN         3\n",
       "33179  chr2  85400441 NaN       A        G  228.171005     NaN         3\n",
       "33346  chr2  85776687 NaN       T        A  115.749001     NaN         1\n",
       "33665  chr2  85918029 NaN       T        C  118.149002     NaN         1\n",
       "34023  chr2  86198668 NaN       A        G  228.250000     NaN         3\n",
       "34039  chr2  86236802 NaN      CT      CTT  228.218994     NaN         3\n",
       "34510  chr2  86658391 NaN       C       CG  174.298996     NaN         3\n",
       "34657  chr2  87049235 NaN       G        T  102.330002     NaN         3"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Identifying mutations caused by CRISPR:\n",
    "##table shows the filtered dataframe with mutations and the quality of the reads as well as the position of mutations\n",
    "\n",
    "intended_sites = [36937210, 36996899, 85400441, 85776687, 85918029, 86198668, 86236802, 86658391, 87049235]\n",
    "\n",
    "intended_mutations_df = filtered_df[df['POS'].isin(intended_sites)]\n",
    "intended_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 444
    },
    "id": "13iweZ4PiDYg",
    "outputId": "bc09304e-2973-4cb1-e5a1-fd2408232ebd"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-17-ae1de7b38eeb>:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  off_target_mutations_df = filtered_df[~df['POS'].isin(intended_sites)]\n"
     ]
    },
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"off_target_mutations_df\",\n  \"rows\": 17752,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57472313,\n        \"min\": 3051696,\n        \"max\": 182002048,\n        \"num_unique_values\": 17750,\n        \"samples\": [\n          5603840\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 224,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 265,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54.76860613909442,\n        \"min\": 60.00130081176758,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 13550,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "off_target_mutations_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-337ef744-ec3a-448e-886e-c2f5a0de70a5\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051696</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TGTCC</td>\n",
       "      <td>96.563599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>111.313004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052386</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>129.570007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052401</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>131.856003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>84.802002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70671</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>95.972900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70672</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>71.400002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70673</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>65.766403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70675</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>77.494499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70677</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>91.032799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17752 rows  8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-337ef744-ec3a-448e-886e-c2f5a0de70a5')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-337ef744-ec3a-448e-886e-c2f5a0de70a5 button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-337ef744-ec3a-448e-886e-c2f5a0de70a5');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-39d5adef-bf34-483c-9728-34eb2e1cbba3\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-39d5adef-bf34-483c-9728-34eb2e1cbba3')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-39d5adef-bf34-483c-9728-34eb2e1cbba3 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_dd36cdf5-1699-4e68-92a0-57233f47b0fd\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('off_target_mutations_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_dd36cdf5-1699-4e68-92a0-57233f47b0fd button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('off_target_mutations_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF    ALT        QUAL  FILTER  GENOTYPE\n",
       "6      chr2    3051696 NaN   T  TGTCC   96.563599     NaN         1\n",
       "29     chr2    3052361 NaN   G      T  111.313004     NaN         1\n",
       "31     chr2    3052386 NaN   T      G  129.570007     NaN         1\n",
       "32     chr2    3052401 NaN   C      G  131.856003     NaN         1\n",
       "33     chr2    3052429 NaN   A      G   84.802002     NaN         1\n",
       "...     ...        ...  ..  ..    ...         ...     ...       ...\n",
       "70671  chr2  182002007 NaN   A      G   95.972900     NaN         1\n",
       "70672  chr2  182002013 NaN   T      C   71.400002     NaN         1\n",
       "70673  chr2  182002021 NaN   G      A   65.766403     NaN         1\n",
       "70675  chr2  182002039 NaN   A      G   77.494499     NaN         1\n",
       "70677  chr2  182002048 NaN   T      C   91.032799     NaN         1\n",
       "\n",
       "[17752 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Detecting Off-Target Effects caused by the CRISPR:\n",
    "#table shows all off target effects created and the position they are located\n",
    "off_target_mutations_df = filtered_df[~df['POS'].isin(intended_sites)]\n",
    "off_target_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 109
    },
    "id": "UZfcP6ejiECM",
    "outputId": "2f2709aa-1244-4963-a5ec-8e35b163a908"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"summary_df\",\n  \"rows\": 1,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Mutation Count\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 17752,\n        \"max\": 17752,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          17752\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Genotype Count\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 17750,\n        \"max\": 17750,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          17750\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "summary_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-f6ae5519-6364-492a-9b0a-f97e05e25b49\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation Count</th>\n",
       "      <th>Genotype Count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr2</th>\n",
       "      <td>17752</td>\n",
       "      <td>17750</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-f6ae5519-6364-492a-9b0a-f97e05e25b49')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-f6ae5519-6364-492a-9b0a-f97e05e25b49 button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-f6ae5519-6364-492a-9b0a-f97e05e25b49');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "  <div id=\"id_869be69e-df66-4a03-a695-a22ab330d804\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('summary_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_869be69e-df66-4a03-a695-a22ab330d804 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('summary_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "       Mutation Count  Genotype Count\n",
       "CHROM                                \n",
       "chr2            17752           17750"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Summarizing Off-Target Mutations\n",
    "summary_df = off_target_mutations_df.groupby('CHROM')['POS'].agg(['count', 'nunique'])\n",
    "summary_df.columns = ['Mutation Count', 'Genotype Count']\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 471
    },
    "id": "T-xOKZg0iIot",
    "outputId": "03cfa1c1-6889-4192-ecc6-606e1aa88ce1"
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAkQAAAHHCAYAAABeLEexAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAABYDklEQVR4nO3deVhUZf8/8PcAzrCDCLIk4Q5iuFEi5pooLrmUlbtouGSouWb2lIJWbqWmYraomLmn6TfzUdBUXMidcMUNlxIkNxA1FPj8/vDHeTwCyuAMi/N+Xddccu5zzzmfe84MvD3baEREQERERGTCzEq6ACIiIqKSxkBEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEZdaBAwfQuHFj2NjYQKPRID4+HgCwefNm1KtXD5aWltBoNLh161aJ1kn6i4qKgkajwYULF0q6lDKjcuXKeP3110u6DKIyi4GISo3jx4+jd+/eeOGFF6DT6eDh4YFevXrh+PHjefo+ePAAb7/9Nm7cuIFZs2Zh6dKl8PLywvXr1/HOO+/AysoKkZGRWLp0KWxsbJTntWjRAhqN5qmP8PDwYhz50125cgXh4eFK6Hua3ECh0Wiwe/fuPPNFBJ6entBoNEX+I7pp06Znfp2++OILrF+//pmW8by7evUqxowZAx8fH1hbW8PGxgb+/v747LPPGPbLkHXr1qFbt26oWrUqrK2t4e3tjdGjR3MbliIafpcZlQbr1q1Djx494OTkhNDQUFSpUgUXLlzAwoULcf36daxcuRJvvPGG0v/UqVOoVasWvv/+ewwYMEBp37x5M9q1a4eYmBgEBQXlWU9MTAyuXr2qTB84cABz5szBxx9/jFq1aintderUQZ06dYw0Wv0dPHgQr7zyChYvXox+/fo9tX9UVBT69+8PS0tL9O/fH/Pnz1fN37FjB1q2bAmdToegoCBs3LhR75qGDh2KyMhIPMuvEFtbW7z11luIiopStWdnZ+PBgwfQ6XTQaDRFXn5Zd+DAAbRv3x4ZGRno3bs3/P39ATx8P6xcuRKNGzdGdHQ0gId7iF566aUibUsyPmdnZ3h4eKBLly548cUXcfToUSxYsABVq1bF4cOHYWVlVdIlmjyLki6A6Ny5c+jTpw+qVq2K2NhYuLi4KPM++OADNG3aFH369EFCQgKqVq0KAEhNTQUAODo6qpZVUHuu1q1bq6YtLS0xZ84ctG7dGi1atHjmsdy5c0e1R6qktW/fHmvWrMGcOXNgYfG/j/vy5cvh7++Pa9eulWB1BTM3N4e5uXlJl1Gibt26hTfeeAPm5uY4cuQIfHx8VPM///xzfP/99wZZV2l73z6Pfv755zy/Y/z9/RESEoJly5ap/mNHJUSIStjgwYMFgMTGxuY7f+fOnQJABg8eLCIiISEhAkD1aN68uTRv3jxPe0hIyBPXvWbNGgEg27dvV9piY2PlrbfeEk9PT9FqtVKpUiUZMWKE3L17V/XckJAQsbGxkbNnz0q7du3E1tZWOnfuLCIid+/elWHDhkmFChXE1tZWOnbsKH/99ZcAkIkTJ6qW89dff0n//v2lYsWKotVqxdfXVxYuXKjM3759e55xAZDFixcXOK7FixcLAFmzZo1oNBrZtGmTMi8zM1PKly8vX331lXh5eUmHDh3yrOvR10NEJCkpSbXO/LbBo79OZsyYIYGBgeLk5CSWlpbSoEEDWbNmjWqZ+T0/d3vl1p+UlKR6TmRkpPj6+opWqxV3d3d5//335ebNm6o+zZs3l9q1a8vx48elRYsWYmVlJR4eHjJt2rQ8r9OcOXPE19dXrKysxNHRUfz9/WXZsmUFvq65rl69Ku+++65UrFhRdDqd1KlTR6KiovJ9zWbMmCHffvutVK1aVbRarbz88suyf//+p65j6tSpAqBQ9YiIsi137dolr7zyiuh0OqlSpYosWbJE1S/3td2xY4cMGTJEXFxcxNHRUZmvz2v8559/SrNmzcTKykqqVaumbOMdO3ZIw4YNxdLSUmrWrCkxMTF56j18+LC0bdtW7OzsxMbGRl577TWJi4tT9bl//76Eh4dL9erVRafTiZOTk7z66qsSHR2t6rdt2zZp0qSJWFtbi4ODg3Tq1ElOnDih6jNx4kQBIImJidKrVy+xt7cXZ2dn+eSTTyQnJ0cuXboknTp1Ejs7O3F1dZUvv/wyT83//vuvTJgwQapVq6b8bhg7dqz8+++/T99A+UhPTxcAMmrUqCI9nwyLgYhKnIeHh1SuXPmJfSpXriyVKlUSEZG9e/fKxx9/LABk+PDhsnTpUomOjpbo6GgZNGiQAJBJkybJ0qVLZe/evU9cbn6BaNiwYdK+fXv54osv5Ntvv5XQ0FAxNzeXt956S/XckJAQ0el0Uq1aNQkJCZEFCxbIjz/+KCIi77zzjgCQPn36SGRkpLzzzjtSt27dPIEoJSVFKlWqJJ6enjJp0iT55ptvpFOnTgJAZs2apfSZNGmSAJBBgwbJ0qVLZenSpXLu3LkCx5X7R+/AgQPSuHFj6dOnjzJv/fr1YmZmJn///XeRA9HevXuldevWAkCpZ+nSpUr/SpUqyfvvvy/z5s2TmTNnSsOGDQWAbNy4UemzdOlS0el00rRpU+X5udsrv0CU+wctKChI5s6dK0OHDhVzc3N55ZVX5P79+0q/5s2bi4eHh3h6esoHH3wg8+fPl9dee00AqILhd999JwDkrbfekm+//Va+/vprCQ0NleHDhxf4uoo8DLu1atWScuXKyciRI2XOnDnStGlTASCzZ8/O85rVr19fqlevLtOmTZPp06eLs7OzVKpUSVVzfho3bixWVlaSmZn5xH65vLy8xNvbW1xdXeXjjz+WefPmSYMGDUSj0cixY8eUfrmvra+vrzRv3lzmzp0rU6dOLfJrPHbsWJk7d674+vqKubm5rFy5Utzc3CQ8PFxmz54tL7zwgjg4OEh6erry/GPHjomNjY24u7vL5MmTZerUqVKlShXR6XTyxx9/KP0+/vhj0Wg0MnDgQPn+++/lq6++kh49eij1iojExMSIhYWF1KxZU6ZPny4RERHi7Ows5cuXz/f9U69ePenRo4fMnz9fOnToIABk5syZ4u3tLUOGDJH58+fLq6++KgBk586dyvOzs7OlTZs2Ym1tLSNGjJBvv/1Whg4dKhYWFsp/hPR1+vRpASBffPFFkZ5PhsVARCXq1q1bAuCpv1ByQ0LuL9XcP9yP73V4NAgURn6B6PE9QSIiU6ZMEY1GIxcvXlTacveSfPTRR6q+hw4dEgAyYsQIVXu/fv3yBKLQ0FBxd3eXa9euqfp2795dHBwclFoOHDjw1L1Cj3r0dZg3b57Y2dkpy3r77belZcuWIiJFDkQiImFhYaq9Qo96/DW8f/++vPTSS/Laa6+p2m1sbPLdi/d4IEpNTRWtVitt2rSR7Oxspd+8efMEgCxatEhpy91TmBtORR7uFXNzc5OuXbsqbZ07d5batWvnW/+TzJ49WwDITz/9pBpfYGCg2NraKu/R3NesQoUKcuPGDaXvhg0bBID8+uuvT1xP+fLlpW7duoWuy8vLK8+e1tTUVNHpdDJ69GilLfe1bdKkiWRlZan66vsaL1++XGk7deqUABAzMzNVqNmyZUue906XLl1Eq9WqQv2VK1fEzs5OmjVrprTVrVtX9f7MT7169aRixYpy/fp1pe3PP/8UMzMz6du3r9KWG4gGDRqktGVlZUmlSpVEo9GoQtbNmzfFyspK9d5cunSpmJmZya5du1TrX7BggQCQPXv2PLHO/OT+Z+v06dN6P5cMj1eZUYm6ffs2AMDOzu6J/XLnp6enG72mR09uvHPnDq5du4bGjRtDRHDkyJE8/YcMGaKa3rx5MwDg/fffV7UPGzZMNS0iWLt2LTp27AgRwbVr15RHcHAw0tLScPjw4WcezzvvvIN79+5h48aNuH37NjZu3IiePXs+83Kf5NHX8ObNm0hLS0PTpk2LPJ6tW7fi/v37GDFiBMzM/vdra+DAgbC3t8dvv/2m6m9ra4vevXsr01qtFg0bNsT58+eVNkdHR/z11184cOCAXrVs2rQJbm5u6NGjh9JWrlw5DB8+HBkZGdi5c6eqf7du3VC+fHllumnTpgCgqiU/6enpT/1cPM7X11dZPgC4uLjA29s733UNHDhQdZ5WUV7j7t27K9Pe3t5wdHRErVq1EBAQoLTn/pxbQ3Z2NqKjo9GlSxflnEAAcHd3R8+ePbF7927lc+7o6Ijjx4/jzJkz+Y43OTkZ8fHx6NevH5ycnJT2OnXqoHXr1ti0aVOe5zx6ro65uTlefvlliAhCQ0OVdkdHxzyv25o1a1CrVi34+PioPquvvfYaAGD79u351liQ5cuXY+HChRg9ejRq1Kih13PJOBiIqETl/sLPDUYFKWxwety9e/eQkpKiejzNpUuXlF+wtra2cHFxQfPmzQEAaWlpqr4WFhaoVKmSqu3ixYswMzNDlSpVVO3Vq1dXTf/zzz+4desWvvvuO7i4uKge/fv3B/C/k8SfhYuLC4KCgrB8+XKsW7cO2dnZeOutt555uU+yceNGNGrUCJaWlnBycoKLiwu++eabPK9fYV28eBHAwz+6j9JqtahataoyP1elSpXyXJ1Wvnx53Lx5U5keN24cbG1t0bBhQ9SoUQNhYWHYs2dPoWqpUaOGKjQAUK5SfLyWF198MU8dAFS15Mfe3v6pn4vHPb6u3PXlt67H35+GeI0dHBzg6emZpw3433j/+ecf3L17N896gIevYU5ODi5fvgwAmDRpEm7duoWaNWvCz88PY8eORUJCwlNrzl3WtWvXcOfOHVX746+Rg4MDLC0t4ezsnKf90dftzJkzOH78eJ7Pas2aNQHo91ndtWsXQkNDERwcjM8//7zQzyPj4lVmVKIcHBzg7u6u+iWXn4SEBLzwwguwt7fXa/mrVq1SwkUuecJl4tnZ2WjdujVu3LiBcePGwcfHBzY2Nvj777/Rr18/5OTkqPrrdLo8fxgLK3dZvXv3RkhISL59DHXpf8+ePTFw4ECkpKSgXbt2BV6FV9Al7tnZ2YVe165du9CpUyc0a9YM8+fPh7u7O8qVK4fFixdj+fLlRSlfbwVdofbotq9VqxYSExOxceNGbN68GWvXrsX8+fMxYcIEREREFGst+fHx8UF8fDzu378PrVZr8HU962XeBa2rqOPNT7NmzXDu3Dls2LAB0dHR+OGHHzBr1iwsWLCgyFdl5VdfYWrOycmBn58fZs6cmW/fx4NgQf7880906tQJL730En7++WfV1Z9UsrglqMS9/vrr+P7777F79240adIkz/xdu3bhwoULGDx4sN7LDg4ORkxMTKH7Hz16FKdPn8aSJUvQt29fpV2fZXh5eSEnJwdJSUmqXeFnz55V9XNxcYGdnR2ys7PzvWfSo571XjxvvPEGBg8ejD/++AOrVq0qsF/u3ovHbxb3+N6BJ9W0du1aWFpaYsuWLdDpdEr74sWLC72Mx3l5eQEAEhMTVYdZ7t+/j6SkpKe+fgWxsbFBt27d0K1bN9y/fx9vvvkmPv/8c4wfPx6WlpYF1pKQkICcnBxVGD516pSq1mfVsWNHxMXFYe3atarDc8ZirNf4cS4uLrC2tkZiYmKeeadOnYKZmZkqXDg5OaF///7o378/MjIy0KxZM4SHh2PAgAGqmvNblrOzs8FuJ1CtWjX8+eefaNWqVZE/j+fOnUPbtm1RsWJFbNq0Cba2tgapjQyDh8yoxI0dOxZWVlYYPHgwrl+/rpp348YNvPfee7C2tsbYsWP1Xra7uzuCgoJUjyfJ/Z/io/8zFBF8/fXXhV5ncHAwAOS5GeLcuXPzrKtr165Yu3Ytjh07lmc5//zzj/Jz7i/1ot7V1tbWFt988w3Cw8PRsWPHAvt5eXnB3NwcsbGxqvbHx/KkmszNzaHRaFR7lS5cuJDvHaltbGwKNaagoCBotVrMmTNHtW0WLlyItLQ0dOjQ4anLeNzj7zWtVgtfX1+ICB48eFDg89q3b4+UlBRVsMzKysLcuXNha2urHF59Vu+99x7c3d0xevRonD59Os/81NRUfPbZZwZZF2Cc1zg/5ubmaNOmDTZs2KD6aparV69i+fLlaNKkibIn+PFtZGtri+rVqyMzMxPAw893vXr1sGTJEtX76NixY4iOjkb79u0NUjPw8Fy8v//+O997P927dy/PobnHpaSkoE2bNjAzM8OWLVtU91uj0oF7iKjE1ahRA0uWLEGvXr3g5+eX507V165dw4oVK1CtWjWj1+Lj44Nq1aphzJgx+Pvvv2Fvb4+1a9c+9XyPR/n7+6Nr166YPXs2rl+/jkaNGmHnzp3KH7VH/3c5depUbN++HQEBARg4cCB8fX1x48YNHD58GFu3bsWNGzcAPPzfqaOjIxYsWAA7OzvY2NggICAgz3kgT1LQYblHOTg44O2338bcuXOh0WhQrVo1bNy4Md/zI3Lvmjx8+HAEBwfD3Nwc3bt3R4cOHTBz5ky0bdsWPXv2RGpqKiIjI1G9evU8h0b9/f2xdetWzJw5Ex4eHqhSpYrqhNxcLi4uGD9+PCIiItC2bVt06tQJiYmJmD9/Pl555RXVCdSF1aZNG7i5ueHVV1+Fq6srTp48iXnz5qFDhw5PPFdt0KBB+Pbbb9GvXz8cOnQIlStXxs8//4w9e/Zg9uzZep/nVpDy5cvjl19+Qfv27VGvXj3VnaoPHz6MFStWIDAw0CDrAozzGhfks88+Q0xMDJo0aYL3338fFhYW+Pbbb5GZmYnp06cr/Xx9fdGiRQv4+/vDyckJBw8exM8//4yhQ4cqfWbMmIF27dohMDAQoaGhuHfvHubOnQsHBweDfgVPnz59sHr1arz33nvYvn07Xn31VWRnZ+PUqVNYvXo1tmzZgpdffrnA57dt2xbnz5/Hhx9+iN27d6u+UsfV1TXPTWOpBJTAlW1E+UpISJAePXqIu7u7lCtXTtzc3KRHjx5y9OjRPH2Nedn9iRMnJCgoSGxtbcXZ2VkGDhwof/75Z55Lh3NvzJifO3fuSFhYmDg5OYmtra106dJFEhMTBYDq8l6Rhzf5CwsLE09PT2XcrVq1ku+++07Vb8OGDeLr6ysWFhaFvjHj016Hxy+7FxH5559/pGvXrmJtbS3ly5eXwYMHy7Fjx/KsMysrS4YNGyYuLi6i0WhUl+AvXLhQatSoITqdTnx8fGTx4sXKZc+POnXqlHJjPxTixozz5s0THx8fKVeunLi6usqQIUMKvGng40JCQsTLy0uZ/vbbb6VZs2ZSoUIF5X5SY8eOlbS0tCe+ZiIPt1n//v3F2dlZtFqt+Pn55dkej96Y8XHI5wadBbly5YqMHDlSatasKZaWlmJtbS3+/v7y+eefq2rNb1uKiHLT0lxPe288y2tcUA0AJCwsTNV2+PBhCQ4OFltbW7G2tpaWLVvmuW/YZ599Jg0bNhRHR0exsrISHx8f+fzzz/Pcw2nr1q3y6quvipWVldjb20vHjh0LvDHjP//8o2ov6HOc3xjv378v06ZNk9q1a4tOp5Py5cuLv7+/REREPPV9g3xuRJr7eHT7UMnhd5kRFZP4+HjUr18fP/30E3r16lXS5RAR0SN4DhGREdy7dy9P2+zZs2FmZoZmzZqVQEVERPQkPIeIyAimT5+OQ4cOoWXLlrCwsMB///tf/Pe//8WgQYMKfXkuEREVHx4yIzKCmJgYRERE4MSJE8jIyMCLL76IPn364D//+Q/vO0JEVAoxEBEREZHJ4zlEREREZPIYiIiIiMjk8WSGQsjJycGVK1dgZ2f3zF+hQERERMVDRHD79m14eHg89XsnGYgK4cqVK7wyiIiIqIy6fPkyKlWq9MQ+DESFkHsr/suXL+v9betERERUMtLT0+Hp6Vmor9RhICqE3MNk9vb2DERERERlTGFOd+FJ1URERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPIuSLoCIiIieb5U/+u2pfS5M7VAMlRSMe4iIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTySjQQxcbGomPHjvDw8IBGo8H69etV8zUaTb6PGTNmKH0qV66cZ/7UqVNVy0lISEDTpk1haWkJT09PTJ8+vTiGR0RERGVEiQaiO3fuoG7duoiMjMx3fnJysuqxaNEiaDQadO3aVdVv0qRJqn7Dhg1T5qWnp6NNmzbw8vLCoUOHMGPGDISHh+O7774z6tiIiIio7LAoyZW3a9cO7dq1K3C+m5ubanrDhg1o2bIlqlatqmq3s7PL0zfXsmXLcP/+fSxatAharRa1a9dGfHw8Zs6ciUGDBj37IIiIiKjMKzPnEF29ehW//fYbQkND88ybOnUqKlSogPr162PGjBnIyspS5sXFxaFZs2bQarVKW3BwMBITE3Hz5s1815WZmYn09HTVg4iIiJ5fJbqHSB9LliyBnZ0d3nzzTVX78OHD0aBBAzg5OWHv3r0YP348kpOTMXPmTABASkoKqlSponqOq6urMq98+fJ51jVlyhREREQYaSRERERU2pSZQLRo0SL06tULlpaWqvZRo0YpP9epUwdarRaDBw/GlClToNPpirSu8ePHq5abnp4OT0/PohVOREREpV6ZCES7du1CYmIiVq1a9dS+AQEByMrKwoULF+Dt7Q03NzdcvXpV1Sd3uqDzjnQ6XZHDFBEREZU9ZeIcooULF8Lf3x9169Z9at/4+HiYmZmhYsWKAIDAwEDExsbiwYMHSp+YmBh4e3vne7iMiIiITE+JBqKMjAzEx8cjPj4eAJCUlIT4+HhcunRJ6ZOeno41a9ZgwIABeZ4fFxeH2bNn488//8T58+exbNkyjBw5Er1791bCTs+ePaHVahEaGorjx49j1apV+Prrr1WHxIiIiMi0leghs4MHD6Jly5bKdG5ICQkJQVRUFABg5cqVEBH06NEjz/N1Oh1WrlyJ8PBwZGZmokqVKhg5cqQq7Dg4OCA6OhphYWHw9/eHs7MzJkyYwEvuiYiISKERESnpIkq79PR0ODg4IC0tDfb29iVdDhERUZlS+aPfntrnwtQOBl+vPn+/y8Q5RERERETGxEBEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyGIiIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZvBINRLGxsejYsSM8PDyg0Wiwfv161fx+/fpBo9GoHm3btlX1uXHjBnr16gV7e3s4OjoiNDQUGRkZqj4JCQlo2rQpLC0t4enpienTpxt7aERERFSGlGggunPnDurWrYvIyMgC+7Rt2xbJycnKY8WKFar5vXr1wvHjxxETE4ONGzciNjYWgwYNUuanp6ejTZs28PLywqFDhzBjxgyEh4fju+++M9q4iIiIqGyxKMmVt2vXDu3atXtiH51OBzc3t3znnTx5Eps3b8aBAwfw8ssvAwDmzp2L9u3b48svv4SHhweWLVuG+/fvY9GiRdBqtahduzbi4+Mxc+ZMVXAiIiIi01XqzyHasWMHKlasCG9vbwwZMgTXr19X5sXFxcHR0VEJQwAQFBQEMzMz7Nu3T+nTrFkzaLVapU9wcDASExNx8+bNfNeZmZmJ9PR01YOIiIieX6U6ELVt2xY//vgjtm3bhmnTpmHnzp1o164dsrOzAQApKSmoWLGi6jkWFhZwcnJCSkqK0sfV1VXVJ3c6t8/jpkyZAgcHB+Xh6elp6KERERFRKVKih8yepnv37srPfn5+qFOnDqpVq4YdO3agVatWRlvv+PHjMWrUKGU6PT2doYiIiOg5Vqr3ED2uatWqcHZ2xtmzZwEAbm5uSE1NVfXJysrCjRs3lPOO3NzccPXqVVWf3OmCzk3S6XSwt7dXPYiIiOj5VaYC0V9//YXr16/D3d0dABAYGIhbt27h0KFDSp/ff/8dOTk5CAgIUPrExsbiwYMHSp+YmBh4e3ujfPnyxTsAIiIiKpVKNBBlZGQgPj4e8fHxAICkpCTEx8fj0qVLyMjIwNixY/HHH3/gwoUL2LZtGzp37ozq1asjODgYAFCrVi20bdsWAwcOxP79+7Fnzx4MHToU3bt3h4eHBwCgZ8+e0Gq1CA0NxfHjx7Fq1Sp8/fXXqkNiREREZNpKNBAdPHgQ9evXR/369QEAo0aNQv369TFhwgSYm5sjISEBnTp1Qs2aNREaGgp/f3/s2rULOp1OWcayZcvg4+ODVq1aoX379mjSpInqHkMODg6Ijo5GUlIS/P39MXr0aEyYMIGX3BMREZFCIyJS0kWUdunp6XBwcEBaWhrPJyIiItJT5Y9+e2qfC1M7GHy9+vz9LlPnEBEREREZAwMRERERmTwGIiIiIjJ5DERERERk8hiIiIiIyOQxEBEREZHJYyAiIiIik8dARERERCaPgYiIiIhMHgMRERERmTwGIiIiIjJ5DERERERk8hiIiIiIyOQxEBEREZHJYyAiIiIik8dARERERCaPgYiIiIhMHgMRERERmTwGIiIiIjJ5DERERERk8vQOREuWLMFvv/2mTH/44YdwdHRE48aNcfHiRYMWR0RERFQc9A5EX3zxBaysrAAAcXFxiIyMxPTp0+Hs7IyRI0cavEAiIiIiY7PQ9wmXL19G9erVAQDr169H165dMWjQILz66qto0aKFoesjIiIiMjq99xDZ2tri+vXrAIDo6Gi0bt0aAGBpaYl79+4ZtjoiIiKiYqD3HqLWrVtjwIABqF+/Pk6fPo327dsDAI4fP47KlSsbuj4iIiIio9N7D1FkZCQCAwPxzz//YO3atahQoQIA4NChQ+jRo4fBCyQiIiIyNr33EDk6OmLevHl52iMiIgxSEBEREVFx0zsQAcCtW7ewf/9+pKamIicnR2nXaDTo06ePwYojIiIiKg56B6Jff/0VvXr1QkZGBuzt7aHRaJR5DERERERUFul9DtHo0aPx7rvvIiMjA7du3cLNmzeVx40bN4xRIxEREZFR6R2I/v77bwwfPhzW1tbGqIeIiIio2OkdiIKDg3Hw4EFj1EJERERUIvQ+h6hDhw4YO3YsTpw4AT8/P5QrV041v1OnTgYrjoiIiKg46B2IBg4cCACYNGlSnnkajQbZ2dnPXhURERFRMdI7ED16mT0RERHR80Dvc4iIiIiInjdFCkQ7d+5Ex44dUb16dVSvXh2dOnXCrl279F5ObGwsOnbsCA8PD2g0Gqxfv16Z9+DBA4wbNw5+fn6wsbGBh4cH+vbtiytXrqiWUblyZWg0GtVj6tSpqj4JCQlo2rQpLC0t4enpienTpxdl2ERERPSc0jsQ/fTTTwgKCoK1tTWGDx+O4cOHw8rKCq1atcLy5cv1WtadO3dQt25dREZG5pl39+5dHD58GJ9++ikOHz6MdevWITExMd+TtidNmoTk5GTlMWzYMGVeeno62rRpAy8vLxw6dAgzZsxAeHg4vvvuO32HTkRERM8pvc8h+vzzzzF9+nSMHDlSaRs+fDhmzpyJyZMno2fPnoVeVrt27dCuXbt85zk4OCAmJkbVNm/ePDRs2BCXLl3Ciy++qLTb2dnBzc0t3+UsW7YM9+/fx6JFi6DValG7dm3Ex8dj5syZGDRoUKFrJSIioueX3nuIzp8/j44dO+Zp79SpE5KSkgxSVEHS0tKg0Wjg6Oioap86dSoqVKiA+vXrY8aMGcjKylLmxcXFoVmzZtBqtUpbcHAwEhMTcfPmTaPWS0RERGWD3nuIPD09sW3bNlSvXl3VvnXrVnh6ehqssMf9+++/GDduHHr06AF7e3ulffjw4WjQoAGcnJywd+9ejB8/HsnJyZg5cyYAICUlBVWqVFEty9XVVZlXvnz5POvKzMxEZmamMp2enm6MIREREVEpoXcgGj16NIYPH474+Hg0btwYALBnzx5ERUXh66+/NniBwMMTrN955x2ICL755hvVvFGjRik/16lTB1qtFoMHD8aUKVOg0+mKtL4pU6YgIiLimWomIiKiskPvQDRkyBC4ubnhq6++wurVqwEAtWrVwqpVq9C5c2eDF5gbhi5evIjff/9dtXcoPwEBAcjKysKFCxfg7e0NNzc3XL16VdUnd7qg847Gjx+vClrp6elG3ftFREREJUvvQAQAb7zxBt544w1D15JHbhg6c+YMtm/fjgoVKjz1OfHx8TAzM0PFihUBAIGBgfjPf/6DBw8eKF8zEhMTA29v73wPlwGATqcr8t4lIiIiKnuKFIgMJSMjA2fPnlWmk5KSEB8fDycnJ7i7u+Ott97C4cOHsXHjRmRnZyMlJQUA4OTkBK1Wi7i4OOzbtw8tW7aEnZ0d4uLiMHLkSPTu3VsJOz179kRERARCQ0Mxbtw4HDt2DF9//TVmzZpVImMmIiKi0kcjIvK0Tk5OTjh9+jScnZ1Rvnx5aDSaAvveuHGj0CvfsWMHWrZsmac9JCQE4eHheU6GzrV9+3a0aNEChw8fxvvvv49Tp04hMzMTVapUQZ8+fTBq1CjVHp6EhASEhYXhwIEDcHZ2xrBhwzBu3LhC15meng4HBwekpaU99ZAdERERqVX+6Len9rkwtYPB16vP3+9C7SGaNWsW7OzslJ+fFIj00aJFCzwpjz0tqzVo0AB//PHHU9dTp06dIt1Jm4iIiExDoQJRSEiI8nO/fv2MVQsRERFRidD7xozm5uZITU3N0379+nWYm5sbpCgiIiKi4qR3ICroMFZmZqbqbtBEREREZUWhrzKbM2cOAECj0eCHH36Ara2tMi87OxuxsbHw8fExfIVERERERlboQJR7mbqIYMGCBarDY1qtFpUrV8aCBQsMXyERERGRkRU6EOV+cWvLli2xbt26Am9qSERERFTW6H1jxu3btxujDiIiIqISU6Q7Vf/111/4v//7P1y6dAn3799Xzcv9lnkiIiKiskLvQLRt2zZ06tQJVatWxalTp/DSSy/hwoULEBE0aNDAGDUSERERGZXel92PHz8eY8aMwdGjR2FpaYm1a9fi8uXLaN68Od5++21j1EhERERkVHoHopMnT6Jv374AAAsLC9y7dw+2traYNGkSpk2bZvACiYiIiIxN70BkY2OjnDfk7u6Oc+fOKfOuXbtmuMqIiIiIione5xA1atQIu3fvRq1atdC+fXuMHj0aR48exbp169CoUSNj1EhERERkVHoHopkzZyIjIwMAEBERgYyMDKxatQo1atTgFWZERERUJukdiKpWrar8bGNjw7tTExERUZmn9zlEVatWxfXr1/O037p1SxWWiIiIiMoKvQPRhQsXkJ2dnac9MzMTf//9t0GKIiIiIipOhT5k9n//93/Kz1u2bIGDg4MynZ2djW3btqFy5coGLY6IiIioOBQ6EHXp0gUAoNFoEBISoppXrlw5VK5cGV999ZVBiyMiIiIqDoUORDk5OQCAKlWq4MCBA3B2djZaUURERETFSe+rzJKSkoxRBxEREVGJ0TsQTZo06YnzJ0yYUORiiIiIiEqC3oHol19+UU0/ePAASUlJsLCwQLVq1RiIiIiIqMzROxAdOXIkT1t6ejr69euHN954wyBFERERERUnve9DlB97e3tERETg008/NcTiiIiIiIqVQQIRAKSlpSEtLc1QiyMiIiIqNnofMpszZ45qWkSQnJyMpUuXol27dgYrjIiIiKi46B2IZs2apZo2MzODi4sLQkJCMH78eIMVRkRERFRceB8iIiIiMnkGO4eIiIiIqKwq9B6id999t1D9Fi1aVORiiIiIiEpCoQNRVFQUvLy8UL9+fYiIMWsiIiIiKlaFDkRDhgzBihUrkJSUhP79+6N3795wcnIyZm1ERERExaLQ5xBFRkYiOTkZH374IX799Vd4enrinXfewZYtW7jHiIiIiMo0vU6q1ul06NGjB2JiYnDixAnUrl0b77//PipXroyMjAxj1UhERERkVEW+yszMzAwajQYiguzsbEPWRERERFSs9ApEmZmZWLFiBVq3bo2aNWvi6NGjmDdvHi5dugRbW1tj1UhERERkVIUORO+//z7c3d0xdepUvP7667h8+TLWrFmD9u3bw8ysaDuaYmNj0bFjR3h4eECj0WD9+vWq+SKCCRMmwN3dHVZWVggKCsKZM2dUfW7cuIFevXrB3t4ejo6OCA0NzXP4LiEhAU2bNoWlpSU8PT0xffr0ItVLREREz6dCX2W2YMECvPjii6hatSp27tyJnTt35ttv3bp1hV75nTt3ULduXbz77rt4880388yfPn065syZgyVLlqBKlSr49NNPERwcjBMnTsDS0hIA0KtXLyQnJyMmJgYPHjxA//79MWjQICxfvhwAkJ6ejjZt2iAoKAgLFizA0aNH8e6778LR0RGDBg0qdK1ERET0/Cp0IOrbty80Go1BV96uXbsCvxBWRDB79mx88skn6Ny5MwDgxx9/hKurK9avX4/u3bvj5MmT2Lx5Mw4cOICXX34ZADB37ly0b98eX375JTw8PLBs2TLcv38fixYtglarRe3atREfH4+ZM2cyEBEREREAPW/MWJySkpKQkpKCoKAgpc3BwQEBAQGIi4tD9+7dERcXB0dHRyUMAUBQUBDMzMywb98+vPHGG4iLi0OzZs2g1WqVPsHBwZg2bRpu3ryJ8uXL51l3ZmYmMjMzlen09HQjjZKIiIhKg1L7XWYpKSkAAFdXV1W7q6urMi8lJQUVK1ZUzbewsICTk5OqT37LeHQdj5syZQocHByUh6en57MPiIiIiEqtUhuIStL48eORlpamPC5fvlzSJREREZERldpA5ObmBgC4evWqqv3q1avKPDc3N6SmpqrmZ2Vl4caNG6o++S3j0XU8TqfTwd7eXvUgIiKi51epDURVqlSBm5sbtm3bprSlp6dj3759CAwMBAAEBgbi1q1bOHTokNLn999/R05ODgICApQ+sbGxePDggdInJiYG3t7e+Z4/RERERKanUIGoQYMGuHnzJgBg0qRJuHv3rkFWnpGRgfj4eMTHxwN4eCJ1fHw8Ll26BI1GgxEjRuCzzz7D//3f/+Ho0aPo27cvPDw80KVLFwBArVq10LZtWwwcOBD79+/Hnj17MHToUHTv3h0eHh4AgJ49e0Kr1SI0NBTHjx/HqlWr8PXXX2PUqFEGGQMRERGVfRopxDezWllZ4cyZM6hUqRLMzc2RnJyc52TmotixYwdatmyZpz0kJARRUVEQEUycOBHfffcdbt26hSZNmmD+/PmoWbOm0vfGjRsYOnQofv31V5iZmaFr166YM2eO6s7ZCQkJCAsLw4EDB+Ds7Ixhw4Zh3Lhxha4zPT0dDg4OSEtL4+EzIiIiPVX+6Len9rkwtYPB16vP3+9CBaLAwEDY2tqiSZMmiIiIwJgxYwr8qo4JEyYUrepSjIGIiIio6MpCICrUfYiioqIwceJEbNy4ERqNBv/9739hYZH3qRqN5rkMRERERPR8K1Qg8vb2xsqVKwE8/Jb7bdu2GeSQGREREVFpUOg7VefKyckxRh1EREREJUbvQAQA586dw+zZs3Hy5EkAgK+vLz744ANUq1bNoMURERERFQe970O0ZcsW+Pr6Yv/+/ahTpw7q1KmDffv2oXbt2oiJiTFGjURERERGpfceoo8++ggjR47E1KlT87SPGzcOrVu3NlhxRERERMVB7z1EJ0+eRGhoaJ72d999FydOnDBIUURERETFSe9A5OLiotxZ+lHx8fG88oyIiIjKJL0PmQ0cOBCDBg3C+fPn0bhxYwDAnj17MG3aNH4dBhEREZVJegeiTz/9FHZ2dvjqq68wfvx4AICHhwfCw8MxfPhwgxdIREREZGx6ByKNRoORI0di5MiRuH37NgDAzs7O4IURERERFZci3YcoF4MQERERPQ/0PqmaiIiI6HnDQEREREQmj4GIiIiITJ5egejBgwdo1aoVzpw5Y6x6iIiIiIqdXoGoXLlySEhIMFYtRERERCVC70NmvXv3xsKFC41RCxEREVGJ0Puy+6ysLCxatAhbt26Fv78/bGxsVPNnzpxpsOKIiIiIioPegejYsWNo0KABAOD06dOqeRqNxjBVERERERUjvQPR9u3bjVEHlTGVP/rtqX0uTO1QDJUQERE9uyLfqfrs2bM4d+4cmjVrBisrK4gI9xAREZFJ4n8Syz69T6q+fv06WrVqhZo1a6J9+/ZITk4GAISGhmL06NEGL5CIiIjI2PQORCNHjkS5cuVw6dIlWFtbK+3dunXD5s2bDVocERERUXHQ+5BZdHQ0tmzZgkqVKqnaa9SogYsXLxqsMCIiIqLiovceojt37qj2DOW6ceMGdDqdQYoiIiIiKk567yFq2rQpfvzxR0yePBnAw0vtc3JyMH36dLRs2dLgBRIRFYQnshKRoegdiKZPn45WrVrh4MGDuH//Pj788EMcP34cN27cwJ49e4xRIxEREZFR6X3I7KWXXsLp06fRpEkTdO7cGXfu3MGbb76JI0eOoFq1asaokYiIiMioinQfIgcHB/znP/8xdC1ERETPLR7iLd2KFIhu3ryJhQsX4uTJkwAAX19f9O/fH05OTgYtjoiIiKg46H3ILDY2FpUrV8acOXNw8+ZN3Lx5E3PmzEGVKlUQGxtrjBqJiIiIjErvPURhYWHo1q0bvvnmG5ibmwMAsrOz8f777yMsLAxHjx41eJFERERExqT3HqKzZ89i9OjRShgCAHNzc4waNQpnz541aHFERERExUHvQNSgQQPl3KFHnTx5EnXr1jVIUURERETFqVCHzBISEpSfhw8fjg8++ABnz55Fo0aNAAB//PEHIiMjMXXqVONUSURERGREhdpDVK9ePdSvXx/16tVDjx49cPnyZXz44Ydo1qwZmjVrhg8//BAXL15Ez549DV5g5cqVodFo8jzCwsIAAC1atMgz77333lMt49KlS+jQoQOsra1RsWJFjB07FllZWQavlYiIiMqmQu0hSkpKMnYdBTpw4ACys7OV6WPHjqF169Z4++23lbaBAwdi0qRJyvSj37WWnZ2NDh06wM3NDXv37kVycjL69u2LcuXK4YsvviieQRAREVGpVqhA5OXlZew6CuTi4qKanjp1KqpVq4bmzZsrbdbW1nBzc8v3+dHR0Thx4gS2bt0KV1dX1KtXD5MnT8a4ceMQHh4OrVZr1PrLosLcPIyIiOh5UqQbM165cgW7d+9GamoqcnJyVPOGDx9ukMLyc//+ffz0008YNWoUNBqN0r5s2TL89NNPcHNzQ8eOHfHpp58qe4ni4uLg5+cHV1dXpX9wcDCGDBmC48ePo379+nnWk5mZiczMTGU6PT3daGMiIiKikqd3IIqKisLgwYOh1WpRoUIFVTDRaDRGDUTr16/HrVu30K9fP6WtZ8+e8PLygoeHBxISEjBu3DgkJiZi3bp1AICUlBRVGAKgTKekpOS7nilTpiAiIsI4gyAiIqJSR+9A9Omnn2LChAkYP348zMz0vmr/mSxcuBDt2rWDh4eH0jZo0CDlZz8/P7i7u6NVq1Y4d+5ckb9sdvz48Rg1apQynZ6eDk9Pz6IXTkRERKWa3oHo7t276N69e7GHoYsXL2Lr1q3Knp+CBAQEAHh4A8lq1arBzc0N+/fvV/W5evUqABR43pFOp4NOpzNA1URERFQW6J1qQkNDsWbNGmPU8kSLFy9GxYoV0aHDk78JOD4+HgDg7u4OAAgMDMTRo0eRmpqq9ImJiYG9vT18fX2NVi8RERGVHXrvIZoyZQpef/11bN68GX5+fihXrpxq/syZMw1WXK6cnBwsXrwYISEhsLD4X8nnzp3D8uXL0b59e1SoUAEJCQkYOXIkmjVrhjp16gAA2rRpA19fX/Tp0wfTp09HSkoKPvnkE4SFhXEvEBEREQEoYiDasmULvL29ASDPSdXGsHXrVly6dAnvvvuuql2r1WLr1q2YPXs27ty5A09PT3Tt2hWffPKJ0sfc3BwbN27EkCFDEBgYCBsbG4SEhKjuW0RERESmTe9A9NVXX2HRokWqK72MrU2bNhCRPO2enp7YuXPnU5/v5eWFTZs2GaM0IiIieg7ofQ6RTqfDq6++aoxaiIiIiEqE3oHogw8+wNy5c41RCxEREVGJ0PuQ2f79+/H7779j48aNqF27dp6Tqp92WTwRERFRaaN3IHJ0dMSbb75pjFqIiIiISoTegWjx4sXGqIOIiIioxBTv7aaJiIiISiG99xBVqVLlifcbOn/+/DMVRERERFTc9A5EI0aMUE0/ePAAR44cwebNmzF27FhD1UVERERUbPQORB988EG+7ZGRkTh48OAzF0RERERU3Ax2DlG7du2wdu1aQy2OiIiIqNgYLBD9/PPPcHJyMtTiiIiIiIqN3ofM6tevrzqpWkSQkpKCf/75B/PnzzdocURERETFQe9A1KVLF9W0mZkZXFxc0KJFC/j4+BiqLiIiIqJio3cgmjhxojHqICIiIioxvDEjERERmbxC7yEyMzN74g0ZAUCj0SArK+uZiyIiIiIqToUORL/88kuB8+Li4jBnzhzk5OQYpCgiIiKi4lToQNS5c+c8bYmJifjoo4/w66+/olevXpg0aZJBiyMiIiIqDkU6h+jKlSsYOHAg/Pz8kJWVhfj4eCxZsgReXl6Gro+IiIjI6PQKRGlpaRg3bhyqV6+O48ePY9u2bfj111/x0ksvGas+IiIiIqMr9CGz6dOnY9q0aXBzc8OKFSvyPYRGREREVBYVOhB99NFHsLKyQvXq1bFkyRIsWbIk337r1q0zWHFERERExaHQgahv375PveyeiIiIqCwqdCCKiooyYhlEREREJYd3qiYiIiKTx0BEREREJo+BiIiIiEweAxERERGZvEKfVE1EZCiVP/rtqX0uTO1QDJVQceO2p9KKe4iIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEwerzIrBXjVBRERUcliIKLnAkMlERE9i1J9yCw8PBwajUb18PHxUeb/+++/CAsLQ4UKFWBra4uuXbvi6tWrqmVcunQJHTp0gLW1NSpWrIixY8ciKyuruIdCREREpVip30NUu3ZtbN26VZm2sPhfySNHjsRvv/2GNWvWwMHBAUOHDsWbb76JPXv2AACys7PRoUMHuLm5Ye/evUhOTkbfvn1Rrlw5fPHFF8U+FiIiIiqdSn0gsrCwgJubW572tLQ0LFy4EMuXL8drr70GAFi8eDFq1aqFP/74A40aNUJ0dDROnDiBrVu3wtXVFfXq1cPkyZMxbtw4hIeHQ6vVFvdwiIiIqBQq1YfMAODMmTPw8PBA1apV0atXL1y6dAkAcOjQITx48ABBQUFKXx8fH7z44ouIi4sDAMTFxcHPzw+urq5Kn+DgYKSnp+P48eMFrjMzMxPp6emqBxERET2/SnUgCggIQFRUFDZv3oxvvvkGSUlJaNq0KW7fvo2UlBRotVo4OjqqnuPq6oqUlBQAQEpKiioM5c7PnVeQKVOmwMHBQXl4enoadmBERERUqpTqQ2bt2rVTfq5Tpw4CAgLg5eWF1atXw8rKymjrHT9+PEaNGqVMp6enMxQRERE9x0p1IHqco6MjatasibNnz6J169a4f/8+bt26pdpLdPXqVeWcIzc3N+zfv1+1jNyr0PI7LymXTqeDTqcz/ACoRPHSfCIiKkipPmT2uIyMDJw7dw7u7u7w9/dHuXLlsG3bNmV+YmIiLl26hMDAQABAYGAgjh49itTUVKVPTEwM7O3t4evrW+z1ExERUelUqvcQjRkzBh07doSXlxeuXLmCiRMnwtzcHD169ICDgwNCQ0MxatQoODk5wd7eHsOGDUNgYCAaNWoEAGjTpg18fX3Rp08fTJ8+HSkpKfjkk08QFhbGPUBERESkKNWB6K+//kKPHj1w/fp1uLi4oEmTJvjjjz/g4uICAJg1axbMzMzQtWtXZGZmIjg4GPPnz1eeb25ujo0bN2LIkCEIDAyEjY0NQkJCMGnSpJIaEhEREZVCpToQrVy58onzLS0tERkZicjIyAL7eHl5YdOmTYYujYiIiJ4jZeocIiIiIiJjYCAiIiIik8dARERERCaPgYiIiIhMHgMRERERmTwGIiIiIjJ5DERERERk8hiIiIiIyOSV6hszkuEV5gtOiYiITA33EBEREZHJ4x4iIiKiJ+CeddPAQERGY6hfIhemdjDIcoieJ4X9fPHzQ1Q4PGRGREREJo+BiIiIiEweD5kRERFRvgpzaPZ5OSzLPURERERk8hiIiIiIyOTxkBkRlUq81Jmo6EzpUJehcA8RERERmTwGIiIiIjJ5DERERERk8ngOERERlSo8/4VKAvcQERERkcljICIiIiKTx0BEREREJo+BiIiIiEweAxERERGZPAYiIiIiMnkMRERERGTyeB8iIio03h+GiJ5XDERERGSy+CXClIuHzIiIiMjkMRARERGRyeMhMyIyKB6CIKKyiHuIiIiIyOQxEBEREZHJK9WHzKZMmYJ169bh1KlTsLKyQuPGjTFt2jR4e3srfVq0aIGdO3eqnjd48GAsWLBAmb506RKGDBmC7du3w9bWFiEhIZgyZQosLEr18ImIigVvp0BUygPRzp07ERYWhldeeQVZWVn4+OOP0aZNG5w4cQI2NjZKv4EDB2LSpEnKtLW1tfJzdnY2OnToADc3N+zduxfJycno27cvypUrhy+++KJYx0NERESlU6kORJs3b1ZNR0VFoWLFijh06BCaNWumtFtbW8PNzS3fZURHR+PEiRPYunUrXF1dUa9ePUyePBnjxo1DeHg4tFqtUcdAREREpV+ZOocoLS0NAODk5KRqX7ZsGZydnfHSSy9h/PjxuHv3rjIvLi4Ofn5+cHV1VdqCg4ORnp6O48ePF0/hREREVKqV6j1Ej8rJycGIESPw6quv4qWXXlLae/bsCS8vL3h4eCAhIQHjxo1DYmIi1q1bBwBISUlRhSEAynRKSkq+68rMzERmZqYynZ6ebujhEFEpwnNoDIO3XKCyrMwEorCwMBw7dgy7d+9WtQ8aNEj52c/PD+7u7mjVqhXOnTuHatWqFWldU6ZMQURExDPVS0RERGVHmThkNnToUGzcuBHbt29HpUqVntg3ICAAAHD27FkAgJubG65evarqkztd0HlH48ePR1pamvK4fPnysw6BiIiISrFSvYdIRDBs2DD88ssv2LFjB6pUqfLU58THxwMA3N3dAQCBgYH4/PPPkZqaiooVKwIAYmJiYG9vD19f33yXodPpoNPpDDMIolKAh4TIFPEQHumjVAeisLAwLF++HBs2bICdnZ1yzo+DgwOsrKxw7tw5LF++HO3bt0eFChWQkJCAkSNHolmzZqhTpw4AoE2bNvD19UWfPn0wffp0pKSk4JNPPkFYWBhDDxEREQEo5YfMvvnmG6SlpaFFixZwd3dXHqtWrQIAaLVabN26FW3atIGPjw9Gjx6Nrl274tdff1WWYW5ujo0bN8Lc3ByBgYHo3bs3+vbtq7pvEREREZm2Ur2HSESeON/T0zPPXarz4+XlhU2bNhmqLCIiInrOlOpARET0vOH5XESlEwMRUSnGP55ERMWDgYjoEQwg9Cx4VRNR2cVA9BzhL2N6Fs/r++d5HReZLr6njYOBiMgIuKeJnjf8I0zPOwYiIqJCYCAger4xEBGVEP6BJSIqPUr1jRmJiIiIigP3EFGpxz0pZGr4nicqftxDRERERCaPgYiIiIhMHgMRERERmTwGIiIiIjJ5PKmaiOg5xhO0iQqHe4iIiIjI5DEQERERkcljICIiIiKTx0BEREREJo+BiIiIiEwerzIrI3ilCBERkfEwEBEREZkg/kdbjYfMiIiIyORxDxGRnvi/KiKi5w/3EBEREZHJYyAiIiIik8dARERERCaPgYiIiIhMHgMRERERmTxeZUZUxvGqNyKiZ8c9RERERGTyuIeIiIiolOAe35LDQERERGUOgwMZGg+ZERERkcljICIiIiKTx0BEREREJo+BiIiIiEyeSQWiyMhIVK5cGZaWlggICMD+/ftLuiQiIiIqBUwmEK1atQqjRo3CxIkTcfjwYdStWxfBwcFITU0t6dKIiIiohGlEREq6iOIQEBCAV155BfPmzQMA5OTkwNPTE8OGDcNHH330xOemp6fDwcEBaWlpsLe3N3htvHyUiIhM3YWpHQy+TH3+fpvEHqL79+/j0KFDCAoKUtrMzMwQFBSEuLi4EqyMiIiISgOTuDHjtWvXkJ2dDVdXV1W7q6srTp06lad/ZmYmMjMzlem0tDQAD5OmMeRk3jXKcomIiMoKY/yNzV1mYQ6GmUQg0teUKVMQERGRp93T07MEqiEiInr+Ocw23rJv374NBweHJ/YxiUDk7OwMc3NzXL16VdV+9epVuLm55ek/fvx4jBo1SpnOycnBjRs3UKFCBWg0Gr3WnZ6eDk9PT1y+fNko5x+VZhw7x86xmw6OnWMvjWMXEdy+fRseHh5P7WsSgUir1cLf3x/btm1Dly5dADwMOdu2bcPQoUPz9NfpdNDpdKo2R0fHZ6rB3t6+VL5ZigPHzrGbGo6dYzc1pXnsT9szlMskAhEAjBo1CiEhIXj55ZfRsGFDzJ49G3fu3EH//v1LujQiIiIqYSYTiLp164Z//vkHEyZMQEpKCurVq4fNmzfnOdGaiIiITI/JBCIAGDp0aL6HyIxJp9Nh4sSJeQ7BmQKOnWM3NRw7x25qnqexm8yNGYmIiIgKYhI3ZiQiIiJ6EgYiIiIiMnkMRERERGTyGIiIiIjI5DEQFUFkZCQqV64MS0tLBAQEYP/+/U/sv2bNGvj4+MDS0hJ+fn7YtGmTar6IYMKECXB3d4eVlRWCgoJw5swZYw6hyPQZ+/fff4+mTZuifPnyKF++PIKCgvL079evHzQajerRtm1bYw+jSPQZe1RUVJ5xWVpaqvo8r9u9RYsWecau0WjQocP/vsm6LGz32NhYdOzYER4eHtBoNFi/fv1Tn7Njxw40aNAAOp0O1atXR1RUVJ4++v7+KAn6jn3dunVo3bo1XFxcYG9vj8DAQGzZskXVJzw8PM829/HxMeIoikbfse/YsSPf93tKSoqq3/O43fP7HGs0GtSuXVvpU1a2O8BApLdVq1Zh1KhRmDhxIg4fPoy6desiODgYqamp+fbfu3cvevTogdDQUBw5cgRdunRBly5dcOzYMaXP9OnTMWfOHCxYsAD79u2DjY0NgoOD8e+//xbXsApF37Hv2LEDPXr0wPbt2xEXFwdPT0+0adMGf//9t6pf27ZtkZycrDxWrFhRHMPRi75jBx7eufXRcV28eFE1/3nd7uvWrVON+9ixYzA3N8fbb7+t6lfat/udO3dQt25dREZGFqp/UlISOnTogJYtWyI+Ph4jRozAgAEDVMGgKO+jkqDv2GNjY9G6dWts2rQJhw4dQsuWLdGxY0ccOXJE1a927dqqbb57925jlP9M9B17rsTERNXYKlasqMx7Xrf7119/rRrz5cuX4eTklOezXha2OwBASC8NGzaUsLAwZTo7O1s8PDxkypQp+fZ/5513pEOHDqq2gIAAGTx4sIiI5OTkiJubm8yYMUOZf+vWLdHpdLJixQojjKDo9B3747KyssTOzk6WLFmitIWEhEjnzp0NXarB6Tv2xYsXi4ODQ4HLM6XtPmvWLLGzs5OMjAylraxs91wA5Jdffnlinw8//FBq166tauvWrZsEBwcr08/6WpaEwow9P76+vhIREaFMT5w4UerWrWu4wopBYca+fft2ASA3b94ssI+pbPdffvlFNBqNXLhwQWkrS9ude4j0cP/+fRw6dAhBQUFKm5mZGYKCghAXF5fvc+Li4lT9ASA4OFjpn5SUhJSUFFUfBwcHBAQEFLjMklCUsT/u7t27ePDgAZycnFTtO3bsQMWKFeHt7Y0hQ4bg+vXrBq39WRV17BkZGfDy8oKnpyc6d+6M48ePK/NMabsvXLgQ3bt3h42Njaq9tG93fT3ts26I17KsyMnJwe3bt/N81s+cOQMPDw9UrVoVvXr1wqVLl0qoQsOrV68e3N3d0bp1a+zZs0dpN6XtvnDhQgQFBcHLy0vVXla2OwORHq5du4bs7Ow8X/fh6uqa53hxrpSUlCf2z/1Xn2WWhKKM/XHjxo2Dh4eH6hdD27Zt8eOPP2Lbtm2YNm0adu7ciXbt2iE7O9ug9T+Loozd29sbixYtwoYNG/DTTz8hJycHjRs3xl9//QXAdLb7/v37cezYMQwYMEDVXha2u74K+qynp6fj3r17BvkMlRVffvklMjIy8M477yhtAQEBiIqKwubNm/HNN98gKSkJTZs2xe3bt0uw0mfn7u6OBQsWYO3atVi7di08PT3RokULHD58GIBhfneWBVeuXMF///vfPJ/1srTdTeqrO6jkTJ06FStXrsSOHTtUJxd3795d+dnPzw916tRBtWrVsGPHDrRq1aokSjWIwMBABAYGKtONGzdGrVq18O2332Ly5MklWFnxWrhwIfz8/NCwYUNV+/O63QlYvnw5IiIisGHDBtV5NO3atVN+rlOnDgICAuDl5YXVq1cjNDS0JEo1CG9vb3h7eyvTjRs3xrlz5zBr1iwsXbq0BCsrXkuWLIGjoyO6dOmiai9L2517iPTg7OwMc3NzXL16VdV+9epVuLm55fscNze3J/bP/VefZZaEoow915dffompU6ciOjoaderUeWLfqlWrwtnZGWfPnn3mmg3lWcaeq1y5cqhfv74yLlPY7nfu3MHKlSsL9UuvNG53fRX0Wbe3t4eVlZVB3kel3cqVKzFgwACsXr06z+HDxzk6OqJmzZplepsXpGHDhsq4TGG7iwgWLVqEPn36QKvVPrFvad7uDER60Gq18Pf3x7Zt25S2nJwcbNu2TbU34FGBgYGq/gAQExOj9K9SpQrc3NxUfdLT07Fv374Cl1kSijJ24OGVVJMnT8bmzZvx8ssvP3U9f/31F65fvw53d3eD1G0IRR37o7Kzs3H06FFlXM/7dgce3m4iMzMTvXv3fup6SuN219fTPuuGeB+VZitWrED//v2xYsUK1S0WCpKRkYFz586V6W1ekPj4eGVcz/t2B4CdO3fi7NmzhfrPT6ne7iV9VndZs3LlStHpdBIVFSUnTpyQQYMGiaOjo6SkpIiISJ8+feSjjz5S+u/Zs0csLCzkyy+/lJMnT8rEiROlXLlycvToUaXP1KlTxdHRUTZs2CAJCQnSuXNnqVKlity7d6/Yx/ck+o596tSpotVq5eeff5bk5GTlcfv2bRERuX37towZM0bi4uIkKSlJtm7dKg0aNJAaNWrIv//+WyJjLIi+Y4+IiJAtW7bIuXPn5NChQ9K9e3extLSU48ePK32e1+2eq0mTJtKtW7c87WVlu9++fVuOHDkiR44cEQAyc+ZMOXLkiFy8eFFERD766CPp06eP0v/8+fNibW0tY8eOlZMnT0pkZKSYm5vL5s2blT5Pey1LC33HvmzZMrGwsJDIyEjVZ/3WrVtKn9GjR8uOHTskKSlJ9uzZI0FBQeLs7CypqanFPr4n0Xfss2bNkvXr18uZM2fk6NGj8sEHH4iZmZls3bpV6fO8bvdcvXv3loCAgHyXWVa2u4gIA1ERzJ07V1588UXRarXSsGFD+eOPP5R5zZs3l5CQEFX/1atXS82aNUWr1Urt2rXlt99+U83PycmRTz/9VFxdXUWn00mrVq0kMTGxOIaiN33G7uXlJQDyPCZOnCgiInfv3pU2bdqIi4uLlCtXTry8vGTgwIGl7pdELn3GPmLECKWvq6urtG/fXg4fPqxa3vO63UVETp06JQAkOjo6z7LKynbPvZz68UfuWENCQqR58+Z5nlOvXj3RarVStWpVWbx4cZ7lPum1LC30HXvz5s2f2F/k4S0I3N3dRavVygsvvCDdunWTs2fPFu/ACkHfsU+bNk2qVasmlpaW4uTkJC1atJDff/89z3Kfx+0u8vB2IVZWVvLdd9/lu8yyst1FRDQiIkbeCUVERERUqvEcIiIiIjJ5DERERERk8hiIiIiIyOQxEBEREZHJYyAiIiIik8dARERERCaPgYiIiIhMHgMRERnFjh07oNFocOvWrSf2q1y5MmbPnm3UWqKiouDo6GjUdRBR0cTGxqJjx47w8PCARqPB+vXr9V7Gli1b0KhRI9jZ2cHFxQVdu3bFhQsX9FoGAxGRCevXrx80Gg00Gg20Wi2qV6+OSZMmISsr65mX3bhxYyQnJ8PBwQFAwaHkwIEDGDRo0DOv70m6deuG06dPP/NyRATff/89AgMDYW9vD1tbW9SuXRsffPBBqfyySqKy4M6dO6hbty4iIyOL9PykpCR07twZr732GuLj47FlyxZcu3YNb775pl7LYSAiMnFt27ZFcnIyzpw5g9GjRyM8PBwzZsx45uVqtVq4ublBo9E8sZ+Liwusra2feX1PYmVlhYoVKz7TMkQEPXv2xPDhw9G+fXtER0fjxIkTWLhwISwtLfHZZ58ZqFoi09KuXTt89tlneOONN/Kdn5mZiTFjxuCFF16AjY0NAgICsGPHDmX+oUOHkJ2djc8++wzVqlVDgwYNMGbMGMTHx+PBgweFL6RkvzmEiEpSSEiIdO7cWdXWunVradSokYiI3LhxQ/r06SOOjo5iZWUlbdu2ldOnTyt9L1y4IK+//ro4OjqKtbW1+Pr6Kt/Vl/u9SDdv3sz3O5Jyv9POy8tLZs2apSzz4sWL0qlTJ7GxsRE7Ozt5++23Vd9zNnHiRKlbt678+OOP4uXlJfb29tKtWzdJT08vcJyLFy8WBweHZ1rGihUrBIBs2LAh3/k5OTmq6e+//158fHxEp9OJt7e3REZGKvOSkpIEgKxdu1ZatGghVlZWUqdOHdm7d69qGT///LP4+vqKVqsVLy8v+fLLL1Xzvby8ZPLkydKnTx+xsbGRF198UTZs2CCpqanKa+jn5ycHDhxQPW/Xrl3SpEkTsbS0lEqVKsmwYcMkIyOjwLETFRcA8ssvv6jaBgwYII0bN5bY2Fg5e/aszJgxQ3Q6nfK76Pz586LVauWHH36QrKwsuXXrlrz99tvSunVr/dZtqEEQUdmTXyDq1KmTNGjQQPm5Vq1aEhsbK/Hx8RIcHCzVq1eX+/fvi4hIhw4dpHXr1pKQkCDnzp2TX3/9VXbu3Cki6kCUmZkps2fPFnt7e+Wb0G/fvi0i6kCUnZ0t9erVkyZNmsjBgwfljz/+EH9/f9UXSk6cOFFsbW3lzTfflKNHj0psbKy4ubnJxx9/XOA48wtE+i6jU6dO4u3tXajX9aeffhJ3d3dZu3atnD9/XtauXStOTk4SFRUlIv8LRD4+PrJx40ZJTEyUt956S7y8vOTBgwciInLw4EExMzOTSZMmSWJioixevFisrKxUXxjr5eUlTk5OsmDBAjl9+rQMGTJE7O3tpW3btrJ69WpJTEyULl26SK1atZTAdvbsWbGxsZFZs2bJ6dOnZc+ePVK/fn3p169focZGZEyPB6KLFy+Kubm5/P3336p+rVq1kvHjxyvTO3bskIoVK4q5ubkAkMDAQLl586Z+636WwomobHs0EOXk5EhMTIzodDoZM2aMnD59WgDInj17lP7Xrl0TKysrWb16tYiI+Pn5SXh4eL7LfjQQieQNJbkeDUTR0dFibm4uly5dUuYfP35cAMj+/ftF5GGYsba2Vu3NGTt2rAQEBBQ4zvwCkb7L8PHxkU6dOqnaPvjgA7GxsREbGxt54YUXlPZq1arJ8uXLVX0nT54sgYGBIvK/QPTDDz/kGefJkydFRKRnz555/oc7duxY8fX1Vaa9vLykd+/eynRycrIAkE8//VRpi4uLEwCSnJwsIiKhoaEyaNAg1XJ37dolZmZmcu/evQLHT1QcHg9EGzduFADK5yz3YWFhIe+8846IPHzf16hRQ8aOHSuHDx+WnTt3SvPmzaVVq1Z59tw+iUXhD64R0fNo48aNsLW1xYMHD5CTk4OePXsiPDwc27Ztg4WFBQICApS+FSpUgLe3N06ePAkAGD58OIYMGYLo6GgEBQWha9euqFOnTpFrOXnyJDw9PeHp6am0+fr6wtHRESdPnsQrr7wC4OGVaXZ2dkofd3d3pKam6rUuQyzjP//5D4YOHYp169bhiy++APDwBNFz584hNDQUAwcOVPpmZWUpJ5jnevS1cnd3BwCkpqbCx8cHJ0+eROfOnVX9X331VcyePRvZ2dkwNzfPswxXV1cAgJ+fX5621NRUuLm54c8//0RCQgKWLVum9BER5OTkICkpCbVq1dLrNSAypoyMDJibm+PQoUPKez6Xra0tACAyMhIODg6YPn26Mu+nn36Cp6cn9u3bh0aNGhVqXQxERCauZcuW+Oabb6DVauHh4QELi8L/WhgwYACCg4Px22+/ITo6GlOmTMFXX32FYcOGGbFioFy5cqppjUaDnJwcoy6jRo0aSExMVLW5uLjAxcVFdcJ2RkYGAOD7779XhUkAeX6hP1pD7snnzzKO3GU8abkZGRkYPHgwhg8fnmdZL774ol7rJjK2+vXrIzs7G6mpqWjatGm+fe7evQszM/U1YrmfNX0+T7zKjMjE2djYoHr16njxxRdVYahWrVrIysrCvn37lLbr168jMTERvr6+Spunpyfee+89rFu3DqNHj8b333+f73q0Wi2ys7OfWEutWrVw+fJlXL58WWk7ceIEbt26pVpnSejRowcSExOxYcOGJ/ZzdXWFh4cHzp8/j+rVq6seVapUKfT6atWqhT179qja9uzZg5o1a+YJVvpo0KABTpw4kae26tWrQ6vVFnm5REWVkZGB+Ph4xMfHA3h4GX18fDwuXbqEmjVrolevXujbty/WrVuHpKQk7N+/H1OmTMFvv/0GAOjQoQMOHDiASZMm4cyZMzh8+DD69+8PLy8v1K9fv9B1MBARUb5q1KiBzp07Y+DAgdi9ezf+/PNP9O7dGy+88IJyKGfEiBHYsmULkpKScPjwYWzfvr3AQy6VK1dGRkYGtm3bhmvXruHu3bt5+gQFBcHPzw+9evXC4cOHsX//fvTt2xfNmzfHyy+/bNTxPk337t3x1ltvoXv37pg0aRL27duHCxcuYOfOnVi1apUqpERERGDKlCmYM2cOTp8+jaNHj2Lx4sWYOXNmodc3evRobNu2DZMnT8bp06exZMkSzJs3D2PGjHmmcYwbNw579+7F0KFDER8fjzNnzmDDhg0YOnToMy2XqKgOHjyI+vXrK+Fl1KhRqF+/PiZMmAAAWLx4Mfr27YvRo0fD29sbXbp0wYEDB5Q9mq+99hqWL1+O9evXo379+mjbti10Oh02b94MKyurQtfBQEREBVq8eDH8/f3x+uuvIzAwECKCTZs2KYdksrOzERYWhlq1aqFt27aoWbMm5s+fn++yGjdujPfeew/dunWDi4uL6nh/Lo1Ggw0bNqB8+fJo1qwZgoKCULVqVaxatcqo4ywMjUaDVatWYfbs2di0aRNatWoFb29vvPvuu/D09MTu3buVvgMGDMAPP/yAxYsXw8/PD82bN0dUVJRee4gaNGiA1atXY+XKlXjppZcwYcIETJo0Cf369XumcdSpUwc7d+7E6dOn0bRpU+UPj4eHxzMtl6ioWrRoAXl4kZfqERUVBeDhIeCIiAgkJSXh/v37uHLlCtatW6c6V6579+44fPgwMjIykJqaig0bNsDHx0evOjT//6xuIiIiIpPFPURERERk8hiIiIiIyOQxEBEREZHJYyAiIiIik8dARERERCaPgYiIiIhMHgMRERERmTwGIiIiIjJ5DERERERk8hiIiIiIyOQxEBEREZHJYyAiIiIik/f/AF5FkxKKSPL9AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#creates a graph with the following qualifications\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "plt.hist(off_target_mutations_df['POS'], bins=50)\n",
    "plt.xlabel('Position in Genome')\n",
    "plt.ylabel('Number of Mutations')\n",
    "plt.title('Off-Target Mutations on Chromosome 2')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 444
    },
    "id": "hxmDpq3xiI6s",
    "outputId": "fae7508f-27bb-4431-bd94-a38fa7adc29a"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-20-f2c3894c7f97>:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  off_target_mutations_df = filtered_df[~df['POS'].isin(intended_sites)]\n"
     ]
    },
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"off_target_mutations_df\",\n  \"rows\": 17752,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57472313,\n        \"min\": 3051696,\n        \"max\": 182002048,\n        \"num_unique_values\": 17750,\n        \"samples\": [\n          5603840\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 224,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 265,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54.76860613909442,\n        \"min\": 60.00130081176758,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 13550,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "off_target_mutations_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-4fa5ceec-38c3-4efc-a5a0-0c10a661910c\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051696</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TGTCC</td>\n",
       "      <td>96.563599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>111.313004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052386</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>129.570007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052401</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>131.856003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>84.802002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70671</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>95.972900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70672</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>71.400002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70673</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>65.766403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70675</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>77.494499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70677</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>91.032799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17752 rows  8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4fa5ceec-38c3-4efc-a5a0-0c10a661910c')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-4fa5ceec-38c3-4efc-a5a0-0c10a661910c button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-4fa5ceec-38c3-4efc-a5a0-0c10a661910c');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-89518abe-6190-41ea-9706-8b2c149d9d64\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-89518abe-6190-41ea-9706-8b2c149d9d64')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-89518abe-6190-41ea-9706-8b2c149d9d64 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_97a917a1-6709-4a62-b162-36816d832761\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('off_target_mutations_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_97a917a1-6709-4a62-b162-36816d832761 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('off_target_mutations_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF    ALT        QUAL  FILTER  GENOTYPE\n",
       "6      chr2    3051696 NaN   T  TGTCC   96.563599     NaN         1\n",
       "29     chr2    3052361 NaN   G      T  111.313004     NaN         1\n",
       "31     chr2    3052386 NaN   T      G  129.570007     NaN         1\n",
       "32     chr2    3052401 NaN   C      G  131.856003     NaN         1\n",
       "33     chr2    3052429 NaN   A      G   84.802002     NaN         1\n",
       "...     ...        ...  ..  ..    ...         ...     ...       ...\n",
       "70671  chr2  182002007 NaN   A      G   95.972900     NaN         1\n",
       "70672  chr2  182002013 NaN   T      C   71.400002     NaN         1\n",
       "70673  chr2  182002021 NaN   G      A   65.766403     NaN         1\n",
       "70675  chr2  182002039 NaN   A      G   77.494499     NaN         1\n",
       "70677  chr2  182002048 NaN   T      C   91.032799     NaN         1\n",
       "\n",
       "[17752 rows x 8 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#creates a new DataFrame by filtering out rows from df where the 'POS' column values match any value in the intended sites list\n",
    "off_target_mutations_df = filtered_df[~df['POS'].isin(intended_sites)]\n",
    "off_target_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 140
    },
    "id": "TasaU4DfiJK1",
    "outputId": "a5ec2705-6ed5-4a28-e57b-d1878df4fd59"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"off_target_summary_df\",\n  \"rows\": 3,\n  \"fields\": [\n    {\n      \"column\": \"mutation_count\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 17752.0,\n        \"max\": 17752.0,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          17752.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"genotype_counts\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 8769.538300275563,\n        \"min\": 2675.0,\n        \"max\": 15077.0,\n        \"num_unique_values\": 2,\n        \"samples\": [\n          2675.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "off_target_summary_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-82070a67-c6d1-4dd1-8a67-36c35bc6a72f\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mutation_count</th>\n",
       "      <th>genotype_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr2</th>\n",
       "      <td>17752.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>(chr2, 1)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>15077.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>(chr2, 3)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2675.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-82070a67-c6d1-4dd1-8a67-36c35bc6a72f')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-82070a67-c6d1-4dd1-8a67-36c35bc6a72f button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-82070a67-c6d1-4dd1-8a67-36c35bc6a72f');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-85b328b0-738c-42f4-a887-4d64fbc3a44d\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-85b328b0-738c-42f4-a887-4d64fbc3a44d')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-85b328b0-738c-42f4-a887-4d64fbc3a44d button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_769cd065-64c1-4ee8-94a7-3cd868824faf\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('off_target_summary_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_769cd065-64c1-4ee8-94a7-3cd868824faf button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('off_target_summary_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "           mutation_count  genotype_counts\n",
       "chr2              17752.0              NaN\n",
       "(chr2, 1)             NaN          15077.0\n",
       "(chr2, 3)             NaN           2675.0"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#groups the off target mutations DataFrame by the 'CHROM' column, calculates the total count of mutations and the counts of each genotype\n",
    "off_target_summary_df = off_target_mutations_df.groupby('CHROM').agg(\n",
    "    mutation_count=('POS', 'count'),\n",
    "    genotype_counts=('GENOTYPE', 'value_counts'))\n",
    "off_target_summary_df"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "name": "python3"
  },
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
